{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# script for download and parse pfam domains using interpro API, and secondary structure predictions using mobidb API\n",
    "# https://www.ebi.ac.uk/interpro/api/protein/UniProt/O96013/entry/pfam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os\n",
    "import requests\n",
    "import pyranges as pr\n",
    "import json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "msa_regions_pdbs = pd.read_csv('../datasets/msa_regions_pdbs.tsv', sep= '\\t')\n",
    "to_calculate = pd.read_csv('../datasets/to_calculate_distances.tsv', sep= '\\t')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PDBs no calculated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "828"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_calculate.pdb.nunique() # 828 prepared to calculate interatomic distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = '../datasets/interatomic_distances_both_regions_kd_cre/'\n",
    "pdbs_calculated = []\n",
    "pdbs_empty = []\n",
    "\n",
    "for file in os.listdir(path):\n",
    "    if file.endswith(\".csv\"):\n",
    "        if os.path.getsize(path + file) < 100:\n",
    "            pdbs_empty.append(file)\n",
    "        else:\n",
    "            pdbs_calculated.append(file) # ok!\n",
    "\n",
    "pdbs_calculated = [ i.split(\".\")[0] for i in pdbs_calculated]\n",
    "pdbs_empty = [ i.split(\".\")[0] for i in pdbs_empty]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "pdbs_no_calculated = set(to_calculate.pdb.unique().tolist()) - set(pdbs_calculated)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Save txt with pdbs no calculated\n",
    "# with open('../datasets/pdbs_no_calculated.txt', 'w') as f:\n",
    "#     for pdb in pdbs_no_calculated:\n",
    "#         f.write(\"%s,\" % pdb)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PDBs empty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1ig1', '1jkl', '1jks', '1p4f', '2hz4', '4fif', '4fii', '4l67']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check if they have different chains belonging to the same uniprot\n",
    "pdbs_empty"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pdb</th>\n",
       "      <th>chain</th>\n",
       "      <th>res_beg</th>\n",
       "      <th>res_end</th>\n",
       "      <th>pdb_beg</th>\n",
       "      <th>pdb_end</th>\n",
       "      <th>sp_beg</th>\n",
       "      <th>sp_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A003</td>\n",
       "      <td>6kv9</td>\n",
       "      <td>A</td>\n",
       "      <td>19</td>\n",
       "      <td>326</td>\n",
       "      <td>13</td>\n",
       "      <td>320</td>\n",
       "      <td>13</td>\n",
       "      <td>320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A003</td>\n",
       "      <td>6kvc</td>\n",
       "      <td>A</td>\n",
       "      <td>19</td>\n",
       "      <td>327</td>\n",
       "      <td>13</td>\n",
       "      <td>321</td>\n",
       "      <td>13</td>\n",
       "      <td>321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A009I821</td>\n",
       "      <td>7m4w</td>\n",
       "      <td>R</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A009I821</td>\n",
       "      <td>7m4x</td>\n",
       "      <td>R</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A009I821</td>\n",
       "      <td>7m4y</td>\n",
       "      <td>R</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "      <td>1</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635147</th>\n",
       "      <td>X8CHM4</td>\n",
       "      <td>5o5l</td>\n",
       "      <td>G</td>\n",
       "      <td>30</td>\n",
       "      <td>253</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635148</th>\n",
       "      <td>X8CHM4</td>\n",
       "      <td>5o5l</td>\n",
       "      <td>H</td>\n",
       "      <td>30</td>\n",
       "      <td>253</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635149</th>\n",
       "      <td>X8CHM4</td>\n",
       "      <td>5o5l</td>\n",
       "      <td>I</td>\n",
       "      <td>30</td>\n",
       "      <td>253</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635150</th>\n",
       "      <td>X8CHM4</td>\n",
       "      <td>5o5l</td>\n",
       "      <td>J</td>\n",
       "      <td>30</td>\n",
       "      <td>253</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635151</th>\n",
       "      <td>X8CHM4</td>\n",
       "      <td>5o5l</td>\n",
       "      <td>K</td>\n",
       "      <td>30</td>\n",
       "      <td>253</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "      <td>205</td>\n",
       "      <td>428</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>635152 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           uniprot   pdb chain  res_beg  res_end  pdb_beg  pdb_end  sp_beg  \\\n",
       "0           A0A003  6kv9     A       19      326       13      320      13   \n",
       "1           A0A003  6kvc     A       19      327       13      321      13   \n",
       "2       A0A009I821  7m4w     R        1      109        1      109       1   \n",
       "3       A0A009I821  7m4x     R        1      109        1      109       1   \n",
       "4       A0A009I821  7m4y     R        1      109        1      109       1   \n",
       "...            ...   ...   ...      ...      ...      ...      ...     ...   \n",
       "635147      X8CHM4  5o5l     G       30      253      205      428     205   \n",
       "635148      X8CHM4  5o5l     H       30      253      205      428     205   \n",
       "635149      X8CHM4  5o5l     I       30      253      205      428     205   \n",
       "635150      X8CHM4  5o5l     J       30      253      205      428     205   \n",
       "635151      X8CHM4  5o5l     K       30      253      205      428     205   \n",
       "\n",
       "        sp_end  \n",
       "0          320  \n",
       "1          321  \n",
       "2          109  \n",
       "3          109  \n",
       "4          109  \n",
       "...        ...  \n",
       "635147     428  \n",
       "635148     428  \n",
       "635149     428  \n",
       "635150     428  \n",
       "635151     428  \n",
       "\n",
       "[635152 rows x 9 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# SIFT pdb - Uniprot mapping\n",
    "pdb_uniprot_mapping = pd.read_csv('../datasets/pdb_uniprot_mapping.tsv', sep= '\\t')\n",
    "pdb_uniprot_mapping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pdb</th>\n",
       "      <th>chain</th>\n",
       "      <th>res_beg</th>\n",
       "      <th>res_end</th>\n",
       "      <th>pdb_beg</th>\n",
       "      <th>pdb_end</th>\n",
       "      <th>sp_beg</th>\n",
       "      <th>sp_end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>125690</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fif</td>\n",
       "      <td>A</td>\n",
       "      <td>51</td>\n",
       "      <td>344</td>\n",
       "      <td>296</td>\n",
       "      <td>589</td>\n",
       "      <td>296</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125691</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fif</td>\n",
       "      <td>B</td>\n",
       "      <td>52</td>\n",
       "      <td>344</td>\n",
       "      <td>297</td>\n",
       "      <td>589</td>\n",
       "      <td>297</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125692</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fif</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>49</td>\n",
       "      <td>55</td>\n",
       "      <td>49</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125693</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fif</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>49</td>\n",
       "      <td>55</td>\n",
       "      <td>49</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125697</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fii</td>\n",
       "      <td>A</td>\n",
       "      <td>54</td>\n",
       "      <td>345</td>\n",
       "      <td>299</td>\n",
       "      <td>590</td>\n",
       "      <td>299</td>\n",
       "      <td>590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125698</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4fii</td>\n",
       "      <td>B</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>49</td>\n",
       "      <td>54</td>\n",
       "      <td>49</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125704</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4l67</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>290</td>\n",
       "      <td>300</td>\n",
       "      <td>589</td>\n",
       "      <td>300</td>\n",
       "      <td>589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125705</th>\n",
       "      <td>O96013</td>\n",
       "      <td>4l67</td>\n",
       "      <td>B</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>39</td>\n",
       "      <td>58</td>\n",
       "      <td>39</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132667</th>\n",
       "      <td>P00519</td>\n",
       "      <td>2hz4</td>\n",
       "      <td>A</td>\n",
       "      <td>7</td>\n",
       "      <td>273</td>\n",
       "      <td>234</td>\n",
       "      <td>500</td>\n",
       "      <td>234</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132668</th>\n",
       "      <td>P00519</td>\n",
       "      <td>2hz4</td>\n",
       "      <td>B</td>\n",
       "      <td>6</td>\n",
       "      <td>271</td>\n",
       "      <td>233</td>\n",
       "      <td>498</td>\n",
       "      <td>233</td>\n",
       "      <td>498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>132669</th>\n",
       "      <td>P00519</td>\n",
       "      <td>2hz4</td>\n",
       "      <td>C</td>\n",
       "      <td>7</td>\n",
       "      <td>273</td>\n",
       "      <td>234</td>\n",
       "      <td>500</td>\n",
       "      <td>234</td>\n",
       "      <td>500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362281</th>\n",
       "      <td>P53355</td>\n",
       "      <td>1ig1</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>293</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362283</th>\n",
       "      <td>P53355</td>\n",
       "      <td>1jkl</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>293</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362284</th>\n",
       "      <td>P53355</td>\n",
       "      <td>1jks</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>293</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "      <td>2</td>\n",
       "      <td>294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>362287</th>\n",
       "      <td>P53355</td>\n",
       "      <td>1p4f</td>\n",
       "      <td>A</td>\n",
       "      <td>1</td>\n",
       "      <td>292</td>\n",
       "      <td>2</td>\n",
       "      <td>293</td>\n",
       "      <td>2</td>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       uniprot   pdb chain  res_beg  res_end  pdb_beg  pdb_end  sp_beg  sp_end\n",
       "125690  O96013  4fif     A       51      344      296      589     296     589\n",
       "125691  O96013  4fif     B       52      344      297      589     297     589\n",
       "125692  O96013  4fif     C        1        7       49       55      49      55\n",
       "125693  O96013  4fif     D        1        7       49       55      49      55\n",
       "125697  O96013  4fii     A       54      345      299      590     299     590\n",
       "125698  O96013  4fii     B        1        6       49       54      49      54\n",
       "125704  O96013  4l67     A        1      290      300      589     300     589\n",
       "125705  O96013  4l67     B        4       23       39       58      39      58\n",
       "132667  P00519  2hz4     A        7      273      234      500     234     500\n",
       "132668  P00519  2hz4     B        6      271      233      498     233     498\n",
       "132669  P00519  2hz4     C        7      273      234      500     234     500\n",
       "362281  P53355  1ig1     A        1      293        2      294       2     294\n",
       "362283  P53355  1jkl     A        1      293        2      294       2     294\n",
       "362284  P53355  1jks     A        1      293        2      294       2     294\n",
       "362287  P53355  1p4f     A        1      292        2      293       2     293"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pdb_uniprot_mapping[pdb_uniprot_mapping.pdb.isin(pdbs_empty)]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CREs into groups  \n",
    "try to clasify CREs into the following groups: N-ter, C-ter, intradomain, other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chain_a</th>\n",
       "      <th>pos_kd</th>\n",
       "      <th>aa_a</th>\n",
       "      <th>atom_a</th>\n",
       "      <th>chain_b</th>\n",
       "      <th>pos_cre</th>\n",
       "      <th>aa_b</th>\n",
       "      <th>atom_b</th>\n",
       "      <th>dist</th>\n",
       "      <th>pdb</th>\n",
       "      <th>pdb_chain</th>\n",
       "      <th>pos_cre_uniprot</th>\n",
       "      <th>msa</th>\n",
       "      <th>kd_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A</td>\n",
       "      <td>111</td>\n",
       "      <td>GLY</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>285</td>\n",
       "      <td>ASN</td>\n",
       "      <td>ND2</td>\n",
       "      <td>2.839370</td>\n",
       "      <td>1a06</td>\n",
       "      <td>1a06_A</td>\n",
       "      <td>285</td>\n",
       "      <td>Q63450_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A</td>\n",
       "      <td>111</td>\n",
       "      <td>GLY</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>285</td>\n",
       "      <td>ASN</td>\n",
       "      <td>ND2</td>\n",
       "      <td>2.839370</td>\n",
       "      <td>1a06</td>\n",
       "      <td>1a06_A</td>\n",
       "      <td>285</td>\n",
       "      <td>Q91YS8_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A</td>\n",
       "      <td>113</td>\n",
       "      <td>TYR</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>286</td>\n",
       "      <td>ILE</td>\n",
       "      <td>N</td>\n",
       "      <td>2.824408</td>\n",
       "      <td>1a06</td>\n",
       "      <td>1a06_A</td>\n",
       "      <td>286</td>\n",
       "      <td>Q63450_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A</td>\n",
       "      <td>113</td>\n",
       "      <td>TYR</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>286</td>\n",
       "      <td>ILE</td>\n",
       "      <td>N</td>\n",
       "      <td>2.824408</td>\n",
       "      <td>1a06</td>\n",
       "      <td>1a06_A</td>\n",
       "      <td>286</td>\n",
       "      <td>Q91YS8_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A</td>\n",
       "      <td>107</td>\n",
       "      <td>ILE</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>287</td>\n",
       "      <td>HIS</td>\n",
       "      <td>NE2</td>\n",
       "      <td>2.719374</td>\n",
       "      <td>1a06</td>\n",
       "      <td>1a06_A</td>\n",
       "      <td>287</td>\n",
       "      <td>Q63450_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14478</th>\n",
       "      <td>A</td>\n",
       "      <td>849</td>\n",
       "      <td>GLN</td>\n",
       "      <td>NE2</td>\n",
       "      <td>A</td>\n",
       "      <td>990</td>\n",
       "      <td>PRO</td>\n",
       "      <td>O</td>\n",
       "      <td>3.034817</td>\n",
       "      <td>8d76</td>\n",
       "      <td>8d76_A</td>\n",
       "      <td>990</td>\n",
       "      <td>P00533_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14479</th>\n",
       "      <td>A</td>\n",
       "      <td>850</td>\n",
       "      <td>HIS</td>\n",
       "      <td>N</td>\n",
       "      <td>A</td>\n",
       "      <td>992</td>\n",
       "      <td>PRO</td>\n",
       "      <td>CG</td>\n",
       "      <td>7.311846</td>\n",
       "      <td>8d76</td>\n",
       "      <td>8d76_A</td>\n",
       "      <td>992</td>\n",
       "      <td>P00533_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14480</th>\n",
       "      <td>A</td>\n",
       "      <td>848</td>\n",
       "      <td>PRO</td>\n",
       "      <td>CD</td>\n",
       "      <td>A</td>\n",
       "      <td>995</td>\n",
       "      <td>SER</td>\n",
       "      <td>OG</td>\n",
       "      <td>3.173897</td>\n",
       "      <td>8d76</td>\n",
       "      <td>8d76_A</td>\n",
       "      <td>995</td>\n",
       "      <td>P00533_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14481</th>\n",
       "      <td>A</td>\n",
       "      <td>741</td>\n",
       "      <td>PRO</td>\n",
       "      <td>CG</td>\n",
       "      <td>A</td>\n",
       "      <td>996</td>\n",
       "      <td>ASN</td>\n",
       "      <td>O</td>\n",
       "      <td>9.931299</td>\n",
       "      <td>8d76</td>\n",
       "      <td>8d76_A</td>\n",
       "      <td>996</td>\n",
       "      <td>P00533_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14482</th>\n",
       "      <td>A</td>\n",
       "      <td>794</td>\n",
       "      <td>PRO</td>\n",
       "      <td>O</td>\n",
       "      <td>A</td>\n",
       "      <td>998</td>\n",
       "      <td>TYR</td>\n",
       "      <td>CE2</td>\n",
       "      <td>3.888776</td>\n",
       "      <td>8d76</td>\n",
       "      <td>8d76_A</td>\n",
       "      <td>998</td>\n",
       "      <td>P00533_60</td>\n",
       "      <td>n-ter</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14483 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chain_a  pos_kd aa_a atom_a chain_b  pos_cre aa_b atom_b      dist  \\\n",
       "0           A     111  GLY      O       A      285  ASN    ND2  2.839370   \n",
       "1           A     111  GLY      O       A      285  ASN    ND2  2.839370   \n",
       "2           A     113  TYR      O       A      286  ILE      N  2.824408   \n",
       "3           A     113  TYR      O       A      286  ILE      N  2.824408   \n",
       "4           A     107  ILE      O       A      287  HIS    NE2  2.719374   \n",
       "...       ...     ...  ...    ...     ...      ...  ...    ...       ...   \n",
       "14478       A     849  GLN    NE2       A      990  PRO      O  3.034817   \n",
       "14479       A     850  HIS      N       A      992  PRO     CG  7.311846   \n",
       "14480       A     848  PRO     CD       A      995  SER     OG  3.173897   \n",
       "14481       A     741  PRO     CG       A      996  ASN      O  9.931299   \n",
       "14482       A     794  PRO      O       A      998  TYR    CE2  3.888776   \n",
       "\n",
       "        pdb pdb_chain  pos_cre_uniprot        msa kd_location  \n",
       "0      1a06    1a06_A              285  Q63450_60       n-ter  \n",
       "1      1a06    1a06_A              285  Q91YS8_60       n-ter  \n",
       "2      1a06    1a06_A              286  Q63450_60       n-ter  \n",
       "3      1a06    1a06_A              286  Q91YS8_60       n-ter  \n",
       "4      1a06    1a06_A              287  Q63450_60       n-ter  \n",
       "...     ...       ...              ...        ...         ...  \n",
       "14478  8d76    8d76_A              990  P00533_60       n-ter  \n",
       "14479  8d76    8d76_A              992  P00533_60       n-ter  \n",
       "14480  8d76    8d76_A              995  P00533_60       n-ter  \n",
       "14481  8d76    8d76_A              996  P00533_60       n-ter  \n",
       "14482  8d76    8d76_A              998  P00533_60       n-ter  \n",
       "\n",
       "[14483 rows x 14 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_shortest_distances_mapped = pd.read_csv('../datasets/interatomic_distances_both_regions_kd_cre/all_shortest_distances_mapped.tsv', sep= '\\t')\n",
    "all_shortest_distances_mapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "n-ter    565\n",
       "c-ter    116\n",
       "Name: kd_location, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_shortest_distances_mapped[['msa', 'pdb', 'kd_location']].drop_duplicates().kd_location.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get the pfam domains from InterPro API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "interpro_api_url = 'https://www.ebi.ac.uk/interpro/api/protein/UniProt/{}/entry/pfam'\n",
    "data_list = []\n",
    "\n",
    "for accession in msa_regions_pdbs.uniprot.unique():\n",
    "    # construct the API url\n",
    "    api_url = interpro_api_url.format(accession)\n",
    "\n",
    "    # Send the API request\n",
    "    response = requests.get(api_url)\n",
    "    # Check if the request was successful\n",
    "    if response.status_code == 200:\n",
    "        # Save the json response\n",
    "        filename = f'../datasets/domains/json_files/{accession}.json'\n",
    "        with open(filename, 'w') as file:\n",
    "            file.write(response.text)\n",
    "        \n",
    "        json_data = response.json()\n",
    "        entry_subset = json_data.get(\"entry_subset\", [])\n",
    "\n",
    "        for entry in entry_subset:\n",
    "            #entry[\"accession\"] = accession\n",
    "\n",
    "            protein_info = json_data.get(\"metadata\", {})\n",
    "            pfam_info = entry\n",
    "\n",
    "            # Extract the desired information from the protein_info\n",
    "            accession = protein_info.get(\"accession\")\n",
    "            id = protein_info.get(\"id\")\n",
    "            source_organism = protein_info.get(\"source_organism\", {}).get(\"scientificName\")\n",
    "            taxId = protein_info.get(\"source_organism\", {}).get(\"taxId\")\n",
    "            scientificName = protein_info.get(\"source_organism\", {}).get(\"scientificName\")\n",
    "            name = protein_info.get(\"name\")\n",
    "            description = protein_info.get(\"description\")\n",
    "            length = protein_info.get(\"length\")\n",
    "            sequence = protein_info.get(\"sequence\")\n",
    "\n",
    "            # Extract the desired information from the pfam_info\n",
    "            pfam_accession = entry.get(\"accession\")\n",
    "            start = entry.get(\"entry_protein_locations\", [])[0].get(\"fragments\", [])[0].get(\"start\")\n",
    "            end = entry.get(\"entry_protein_locations\", [])[0].get(\"fragments\", [])[0].get(\"end\")\n",
    "            entry_integrated = entry.get(\"entry_integrated\")\n",
    "\n",
    "\n",
    "            # Append the extracted data to the data_list\n",
    "            data_list.append({\n",
    "                \"accession\": accession,\n",
    "                \"id\": id,\n",
    "                \"source_organism\": source_organism,\n",
    "                \"taxId\": taxId,\n",
    "                \"scientificName\": scientificName,\n",
    "                \"name\": name,\n",
    "                \"description\": description,\n",
    "                \"length\": length,\n",
    "                \"sequence\": sequence,\n",
    "                \"pfam_accession\": pfam_accession,\n",
    "                \"start\": start,\n",
    "                \"end\": end,\n",
    "                \"entry_integrated\": entry_integrated\n",
    "            })\n",
    "\n",
    "        #data_list.extend(entry_subset)\n",
    "\n",
    "\n",
    "    else:\n",
    "        print(f'Error downloading json for {accession}. Status code: {response.status_code}')\n",
    "\n",
    "df = pd.DataFrame(data_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>id</th>\n",
       "      <th>source_organism</th>\n",
       "      <th>taxId</th>\n",
       "      <th>scientificName</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>length</th>\n",
       "      <th>sequence</th>\n",
       "      <th>pfam_accession</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>entry_integrated</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>KGP_PLAVS</td>\n",
       "      <td>Plasmodium vivax (strain Salvador I)</td>\n",
       "      <td>126793</td>\n",
       "      <td>Plasmodium vivax (strain Salvador I)</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>[Serine/threonine protein kinase which acts as...</td>\n",
       "      <td>846</td>\n",
       "      <td>MRCNERNKKKAIFSNDDFSGEDTLMEDHLQLREKLSEDIEMIKASL...</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>71</td>\n",
       "      <td>151</td>\n",
       "      <td>ipr000595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>KGP_PLAVS</td>\n",
       "      <td>Plasmodium vivax (strain Salvador I)</td>\n",
       "      <td>126793</td>\n",
       "      <td>Plasmodium vivax (strain Salvador I)</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>[Serine/threonine protein kinase which acts as...</td>\n",
       "      <td>846</td>\n",
       "      <td>MRCNERNKKKAIFSNDDFSGEDTLMEDHLQLREKLSEDIEMIKASL...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>535</td>\n",
       "      <td>791</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>KGP_PLAF7</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>36329</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>[Serine/threonine protein kinase which acts as...</td>\n",
       "      <td>853</td>\n",
       "      <td>MEEDDNLKKGNERNKKKAIFSNDDFTGEDSLMEDHLELREKLSEDI...</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>78</td>\n",
       "      <td>158</td>\n",
       "      <td>ipr000595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>KGP_PLAF7</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>36329</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>[Serine/threonine protein kinase which acts as...</td>\n",
       "      <td>853</td>\n",
       "      <td>MEEDDNLKKGNERNKKKAIFSNDDFTGEDSLMEDHLELREKLSEDI...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>542</td>\n",
       "      <td>798</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O74536</td>\n",
       "      <td>SNF1_SCHPO</td>\n",
       "      <td>Schizosaccharomyces pombe (strain 972 / ATCC 2...</td>\n",
       "      <td>284812</td>\n",
       "      <td>Schizosaccharomyces pombe (strain 972 / ATCC 2...</td>\n",
       "      <td>SNF1-like protein kinase ssp2</td>\n",
       "      <td>[Serine/threonine protein kinase essential for...</td>\n",
       "      <td>576</td>\n",
       "      <td>MQPQEVDLMENSTMRNGARVLPPEAISKRHIGPYIIRETLGEGSFG...</td>\n",
       "      <td>PF16579</td>\n",
       "      <td>451</td>\n",
       "      <td>573</td>\n",
       "      <td>ipr032270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>CDPK4_PLAF7</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>36329</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>[Calcium-dependent protein kinase which acts a...</td>\n",
       "      <td>528</td>\n",
       "      <td>MGQEVSSVNNTKNEHHKTNKKSLKGGNERHEMKESSVGISKKIVEN...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>70</td>\n",
       "      <td>328</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>CDPK4_PLAF7</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>36329</td>\n",
       "      <td>Plasmodium falciparum (isolate 3D7)</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>[Calcium-dependent protein kinase which acts a...</td>\n",
       "      <td>528</td>\n",
       "      <td>MGQEVSSVNNTKNEHHKTNKKSLKGGNERHEMKESSVGISKKIVEN...</td>\n",
       "      <td>PF13499</td>\n",
       "      <td>381</td>\n",
       "      <td>450</td>\n",
       "      <td>ipr002048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Q8VDF3</td>\n",
       "      <td>DAPK2_MOUSE</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>10090</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>[Calcium/calmodulin-dependent serine/threonine...</td>\n",
       "      <td>370</td>\n",
       "      <td>MVQASMRSPNMETFKQQKVEDFYDIGEELGSGQFAIVKKCREKSTG...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Q9UIK4</td>\n",
       "      <td>DAPK2_HUMAN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>[Calcium/calmodulin-dependent serine/threonine...</td>\n",
       "      <td>370</td>\n",
       "      <td>MFQASMRSPNMEPFKQQKVEDFYDIGEELGSGQFAIVKKCREKSTG...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Q96NX5</td>\n",
       "      <td>KCC1G_HUMAN</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>[Calcium/calmodulin-dependent protein kinase b...</td>\n",
       "      <td>476</td>\n",
       "      <td>MGRKEEDDCSSWKKQTTNIRKTFIFMEVLGSGAFSEVFLVKQRLTG...</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>ipr000719</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    accession           id                                    source_organism  \\\n",
       "0      A5K0N4    KGP_PLAVS               Plasmodium vivax (strain Salvador I)   \n",
       "1      A5K0N4    KGP_PLAVS               Plasmodium vivax (strain Salvador I)   \n",
       "2      Q8I719    KGP_PLAF7                Plasmodium falciparum (isolate 3D7)   \n",
       "3      Q8I719    KGP_PLAF7                Plasmodium falciparum (isolate 3D7)   \n",
       "4      O74536   SNF1_SCHPO  Schizosaccharomyces pombe (strain 972 / ATCC 2...   \n",
       "..        ...          ...                                                ...   \n",
       "104    Q8IBS5  CDPK4_PLAF7                Plasmodium falciparum (isolate 3D7)   \n",
       "105    Q8IBS5  CDPK4_PLAF7                Plasmodium falciparum (isolate 3D7)   \n",
       "106    Q8VDF3  DAPK2_MOUSE                                       Mus musculus   \n",
       "107    Q9UIK4  DAPK2_HUMAN                                       Homo sapiens   \n",
       "108    Q96NX5  KCC1G_HUMAN                                       Homo sapiens   \n",
       "\n",
       "      taxId                                     scientificName  \\\n",
       "0    126793               Plasmodium vivax (strain Salvador I)   \n",
       "1    126793               Plasmodium vivax (strain Salvador I)   \n",
       "2     36329                Plasmodium falciparum (isolate 3D7)   \n",
       "3     36329                Plasmodium falciparum (isolate 3D7)   \n",
       "4    284812  Schizosaccharomyces pombe (strain 972 / ATCC 2...   \n",
       "..      ...                                                ...   \n",
       "104   36329                Plasmodium falciparum (isolate 3D7)   \n",
       "105   36329                Plasmodium falciparum (isolate 3D7)   \n",
       "106   10090                                       Mus musculus   \n",
       "107    9606                                       Homo sapiens   \n",
       "108    9606                                       Homo sapiens   \n",
       "\n",
       "                                                  name  \\\n",
       "0                        cGMP-dependent protein kinase   \n",
       "1                        cGMP-dependent protein kinase   \n",
       "2                        cGMP-dependent protein kinase   \n",
       "3                        cGMP-dependent protein kinase   \n",
       "4                        SNF1-like protein kinase ssp2   \n",
       "..                                                 ...   \n",
       "104                 Calcium-dependent protein kinase 4   \n",
       "105                 Calcium-dependent protein kinase 4   \n",
       "106                  Death-associated protein kinase 2   \n",
       "107                  Death-associated protein kinase 2   \n",
       "108  Calcium/calmodulin-dependent protein kinase ty...   \n",
       "\n",
       "                                           description  length  \\\n",
       "0    [Serine/threonine protein kinase which acts as...     846   \n",
       "1    [Serine/threonine protein kinase which acts as...     846   \n",
       "2    [Serine/threonine protein kinase which acts as...     853   \n",
       "3    [Serine/threonine protein kinase which acts as...     853   \n",
       "4    [Serine/threonine protein kinase essential for...     576   \n",
       "..                                                 ...     ...   \n",
       "104  [Calcium-dependent protein kinase which acts a...     528   \n",
       "105  [Calcium-dependent protein kinase which acts a...     528   \n",
       "106  [Calcium/calmodulin-dependent serine/threonine...     370   \n",
       "107  [Calcium/calmodulin-dependent serine/threonine...     370   \n",
       "108  [Calcium/calmodulin-dependent protein kinase b...     476   \n",
       "\n",
       "                                              sequence pfam_accession  start  \\\n",
       "0    MRCNERNKKKAIFSNDDFSGEDTLMEDHLQLREKLSEDIEMIKASL...        PF00027     71   \n",
       "1    MRCNERNKKKAIFSNDDFSGEDTLMEDHLQLREKLSEDIEMIKASL...        PF00069    535   \n",
       "2    MEEDDNLKKGNERNKKKAIFSNDDFTGEDSLMEDHLELREKLSEDI...        PF00027     78   \n",
       "3    MEEDDNLKKGNERNKKKAIFSNDDFTGEDSLMEDHLELREKLSEDI...        PF00069    542   \n",
       "4    MQPQEVDLMENSTMRNGARVLPPEAISKRHIGPYIIRETLGEGSFG...        PF16579    451   \n",
       "..                                                 ...            ...    ...   \n",
       "104  MGQEVSSVNNTKNEHHKTNKKSLKGGNERHEMKESSVGISKKIVEN...        PF00069     70   \n",
       "105  MGQEVSSVNNTKNEHHKTNKKSLKGGNERHEMKESSVGISKKIVEN...        PF13499    381   \n",
       "106  MVQASMRSPNMETFKQQKVEDFYDIGEELGSGQFAIVKKCREKSTG...        PF00069     23   \n",
       "107  MFQASMRSPNMEPFKQQKVEDFYDIGEELGSGQFAIVKKCREKSTG...        PF00069     23   \n",
       "108  MGRKEEDDCSSWKKQTTNIRKTFIFMEVLGSGAFSEVFLVKQRLTG...        PF00069     25   \n",
       "\n",
       "     end entry_integrated  \n",
       "0    151        ipr000595  \n",
       "1    791        ipr000719  \n",
       "2    158        ipr000595  \n",
       "3    798        ipr000719  \n",
       "4    573        ipr032270  \n",
       "..   ...              ...  \n",
       "104  328        ipr000719  \n",
       "105  450        ipr002048  \n",
       "106  285        ipr000719  \n",
       "107  285        ipr000719  \n",
       "108  277        ipr000719  \n",
       "\n",
       "[109 rows x 13 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pfam</th>\n",
       "      <th>domain_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>PF00001</td>\n",
       "      <td>7tm_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>PF00002</td>\n",
       "      <td>7tm_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PF00003</td>\n",
       "      <td>7tm_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>PF00004</td>\n",
       "      <td>AAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>155</th>\n",
       "      <td>PF00005</td>\n",
       "      <td>ABC_tran</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7835</th>\n",
       "      <td>PF20620</td>\n",
       "      <td>DUF6805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7836</th>\n",
       "      <td>PF20621</td>\n",
       "      <td>DUF6806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1233</th>\n",
       "      <td>PF20622</td>\n",
       "      <td>Big_15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15497</th>\n",
       "      <td>PF20623</td>\n",
       "      <td>Sgo0707_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3011</th>\n",
       "      <td>PF20624</td>\n",
       "      <td>DMRT5_DMB</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18109 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          pfam domain_name\n",
       "42     PF00001       7tm_1\n",
       "43     PF00002       7tm_2\n",
       "44     PF00003       7tm_3\n",
       "72     PF00004         AAA\n",
       "155    PF00005    ABC_tran\n",
       "...        ...         ...\n",
       "7835   PF20620     DUF6805\n",
       "7836   PF20621     DUF6806\n",
       "1233   PF20622      Big_15\n",
       "15497  PF20623  Sgo0707_N2\n",
       "3011   PF20624   DMRT5_DMB\n",
       "\n",
       "[18109 rows x 2 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add domains name\n",
    "import re\n",
    "import gzip\n",
    "\n",
    "def parse_pfam_mapping(file_path):\n",
    "    accession_domain_mapping = {}\n",
    "\n",
    "    with gzip.open(file_path, 'rt') as file:\n",
    "        content = file.read()\n",
    "\n",
    "    # Use regex to extract accession and domain names\n",
    "    pattern = r\"#=GF ID\\s+(\\w+)\\n#=GF AC\\s+([\\w\\.]+)\\n\"\n",
    "    matches = re.findall(pattern, content)\n",
    "\n",
    "    # Create a dictionary with accession as keys and domain names as values\n",
    "    for match in matches:\n",
    "        accession = match[1]\n",
    "        accession = accession.split(\".\")[0]\n",
    "        domain_name = match[0]\n",
    "        accession_domain_mapping[accession] = domain_name\n",
    "\n",
    "    return accession_domain_mapping\n",
    "\n",
    "# Specify the path to the PFAM mapping file\n",
    "file_path = '../raw_data/Pfam-A.hmm.dat.gz'\n",
    "\n",
    "# Parse the file and get the accession-domain mapping\n",
    "mapping = parse_pfam_mapping(file_path)\n",
    "\n",
    "pfam_domains = pd.DataFrame.from_dict(mapping, orient= 'index', columns= ['domain_name']).reset_index().rename(columns={'index': 'pfam'}).sort_values('pfam')\n",
    "pfam_domains\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pfam</th>\n",
       "      <th>domain_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PF00001</td>\n",
       "      <td>7tm_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PF00002</td>\n",
       "      <td>7tm_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PF00003</td>\n",
       "      <td>7tm_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PF00004</td>\n",
       "      <td>AAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PF00005</td>\n",
       "      <td>ABC_tran</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18105</th>\n",
       "      <td>PF20621</td>\n",
       "      <td>DUF6806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18106</th>\n",
       "      <td>PF20622</td>\n",
       "      <td>Big_15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18107</th>\n",
       "      <td>PF20623</td>\n",
       "      <td>Sgo0707_N2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18108</th>\n",
       "      <td>PF20624</td>\n",
       "      <td>DMRT5_DMB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18109</th>\n",
       "      <td>PF07714</td>\n",
       "      <td>PK_Tyr_Ser</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18110 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          pfam domain_name\n",
       "0      PF00001       7tm_1\n",
       "1      PF00002       7tm_2\n",
       "2      PF00003       7tm_3\n",
       "3      PF00004         AAA\n",
       "4      PF00005    ABC_tran\n",
       "...        ...         ...\n",
       "18105  PF20621     DUF6806\n",
       "18106  PF20622      Big_15\n",
       "18107  PF20623  Sgo0707_N2\n",
       "18108  PF20624   DMRT5_DMB\n",
       "18109  PF07714  PK_Tyr_Ser\n",
       "\n",
       "[18110 rows x 2 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_add = {'pfam': 'PF07714', 'domain_name': 'PK_Tyr_Ser'}\n",
    "pfam_domains = pfam_domains.append(to_add, ignore_index= True)\n",
    "pfam_domains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Format it\n",
    "df = df[['accession', 'source_organism', 'name', 'pfam_accession', 'start', 'end', 'entry_integrated']].rename(\n",
    "    columns= {\n",
    "        'accession': 'uniprot',\n",
    "        'name': 'protein_name',\n",
    "        'pfam_accession': 'pfam',\n",
    "        'entry_integrated': 'interpro'\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pfam</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>domain_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>71</td>\n",
       "      <td>151</td>\n",
       "      <td>cNMP_binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>535</td>\n",
       "      <td>791</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>78</td>\n",
       "      <td>158</td>\n",
       "      <td>cNMP_binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>542</td>\n",
       "      <td>798</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O74536</td>\n",
       "      <td>PF16579</td>\n",
       "      <td>SNF1-like protein kinase ssp2</td>\n",
       "      <td>451</td>\n",
       "      <td>573</td>\n",
       "      <td>AdenylateSensor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>70</td>\n",
       "      <td>328</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>PF13499</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>381</td>\n",
       "      <td>450</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>Q8VDF3</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>Q9UIK4</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>Q96NX5</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>Pkinase</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>109 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    uniprot     pfam                                       protein_name  \\\n",
       "0    A5K0N4  PF00027                      cGMP-dependent protein kinase   \n",
       "1    A5K0N4  PF00069                      cGMP-dependent protein kinase   \n",
       "2    Q8I719  PF00027                      cGMP-dependent protein kinase   \n",
       "3    Q8I719  PF00069                      cGMP-dependent protein kinase   \n",
       "4    O74536  PF16579                      SNF1-like protein kinase ssp2   \n",
       "..      ...      ...                                                ...   \n",
       "104  Q8IBS5  PF00069                 Calcium-dependent protein kinase 4   \n",
       "105  Q8IBS5  PF13499                 Calcium-dependent protein kinase 4   \n",
       "106  Q8VDF3  PF00069                  Death-associated protein kinase 2   \n",
       "107  Q9UIK4  PF00069                  Death-associated protein kinase 2   \n",
       "108  Q96NX5  PF00069  Calcium/calmodulin-dependent protein kinase ty...   \n",
       "\n",
       "     start  end      domain_name  \n",
       "0       71  151     cNMP_binding  \n",
       "1      535  791          Pkinase  \n",
       "2       78  158     cNMP_binding  \n",
       "3      542  798          Pkinase  \n",
       "4      451  573  AdenylateSensor  \n",
       "..     ...  ...              ...  \n",
       "104     70  328          Pkinase  \n",
       "105    381  450              NaN  \n",
       "106     23  285          Pkinase  \n",
       "107     23  285          Pkinase  \n",
       "108     25  277          Pkinase  \n",
       "\n",
       "[109 rows x 6 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam = df[['uniprot', 'pfam', 'protein_name', 'start', 'end']].merge(pfam_domains, how= 'left')\n",
    "pfam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['PF00757', 'PF17988', 'PF07679', 'PF13499'], dtype=object)"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pfam[pfam.domain_name.isna()].pfam.unique() # les falta el domain name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pfam.to_csv('../datasets/pfam_domains.tsv', sep= '\\t', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pfam</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>domain_name</th>\n",
       "      <th>term_id_cre</th>\n",
       "      <th>start_cre</th>\n",
       "      <th>end_cre</th>\n",
       "      <th>length_cre</th>\n",
       "      <th>term_id_kd</th>\n",
       "      <th>start_kd</th>\n",
       "      <th>end_kd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>71</td>\n",
       "      <td>151</td>\n",
       "      <td>cNMP_binding</td>\n",
       "      <td>cre10</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>kd10</td>\n",
       "      <td>534</td>\n",
       "      <td>791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A5K0N4</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>535</td>\n",
       "      <td>791</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre10</td>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>kd10</td>\n",
       "      <td>534</td>\n",
       "      <td>791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>PF00027</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>78</td>\n",
       "      <td>158</td>\n",
       "      <td>cNMP_binding</td>\n",
       "      <td>cre223</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>kd223</td>\n",
       "      <td>541</td>\n",
       "      <td>798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q8I719</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>cGMP-dependent protein kinase</td>\n",
       "      <td>542</td>\n",
       "      <td>798</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre223</td>\n",
       "      <td>1</td>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "      <td>kd223</td>\n",
       "      <td>541</td>\n",
       "      <td>798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O74536</td>\n",
       "      <td>PF16579</td>\n",
       "      <td>SNF1-like protein kinase ssp2</td>\n",
       "      <td>451</td>\n",
       "      <td>573</td>\n",
       "      <td>AdenylateSensor</td>\n",
       "      <td>cre21</td>\n",
       "      <td>305</td>\n",
       "      <td>351</td>\n",
       "      <td>47</td>\n",
       "      <td>kd21</td>\n",
       "      <td>34</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>70</td>\n",
       "      <td>328</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre224</td>\n",
       "      <td>350</td>\n",
       "      <td>358</td>\n",
       "      <td>9</td>\n",
       "      <td>kd224</td>\n",
       "      <td>71</td>\n",
       "      <td>329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>Q8IBS5</td>\n",
       "      <td>PF13499</td>\n",
       "      <td>Calcium-dependent protein kinase 4</td>\n",
       "      <td>381</td>\n",
       "      <td>450</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cre224</td>\n",
       "      <td>350</td>\n",
       "      <td>358</td>\n",
       "      <td>9</td>\n",
       "      <td>kd224</td>\n",
       "      <td>71</td>\n",
       "      <td>329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>Q8VDF3</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre233</td>\n",
       "      <td>292</td>\n",
       "      <td>301</td>\n",
       "      <td>10</td>\n",
       "      <td>kd233</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>Q9UIK4</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Death-associated protein kinase 2</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre273</td>\n",
       "      <td>292</td>\n",
       "      <td>301</td>\n",
       "      <td>10</td>\n",
       "      <td>kd273</td>\n",
       "      <td>23</td>\n",
       "      <td>285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>Q96NX5</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre243</td>\n",
       "      <td>277</td>\n",
       "      <td>317</td>\n",
       "      <td>41</td>\n",
       "      <td>kd243</td>\n",
       "      <td>23</td>\n",
       "      <td>277</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>114 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    uniprot     pfam                                       protein_name  \\\n",
       "0    A5K0N4  PF00027                      cGMP-dependent protein kinase   \n",
       "1    A5K0N4  PF00069                      cGMP-dependent protein kinase   \n",
       "2    Q8I719  PF00027                      cGMP-dependent protein kinase   \n",
       "3    Q8I719  PF00069                      cGMP-dependent protein kinase   \n",
       "4    O74536  PF16579                      SNF1-like protein kinase ssp2   \n",
       "..      ...      ...                                                ...   \n",
       "109  Q8IBS5  PF00069                 Calcium-dependent protein kinase 4   \n",
       "110  Q8IBS5  PF13499                 Calcium-dependent protein kinase 4   \n",
       "111  Q8VDF3  PF00069                  Death-associated protein kinase 2   \n",
       "112  Q9UIK4  PF00069                  Death-associated protein kinase 2   \n",
       "113  Q96NX5  PF00069  Calcium/calmodulin-dependent protein kinase ty...   \n",
       "\n",
       "     start  end      domain_name term_id_cre  start_cre  end_cre  length_cre  \\\n",
       "0       71  151     cNMP_binding       cre10          1       22          22   \n",
       "1      535  791          Pkinase       cre10          1       22          22   \n",
       "2       78  158     cNMP_binding      cre223          1       29          29   \n",
       "3      542  798          Pkinase      cre223          1       29          29   \n",
       "4      451  573  AdenylateSensor       cre21        305      351          47   \n",
       "..     ...  ...              ...         ...        ...      ...         ...   \n",
       "109     70  328          Pkinase      cre224        350      358           9   \n",
       "110    381  450              NaN      cre224        350      358           9   \n",
       "111     23  285          Pkinase      cre233        292      301          10   \n",
       "112     23  285          Pkinase      cre273        292      301          10   \n",
       "113     25  277          Pkinase      cre243        277      317          41   \n",
       "\n",
       "    term_id_kd  start_kd  end_kd  \n",
       "0         kd10       534     791  \n",
       "1         kd10       534     791  \n",
       "2        kd223       541     798  \n",
       "3        kd223       541     798  \n",
       "4         kd21        34     285  \n",
       "..         ...       ...     ...  \n",
       "109      kd224        71     329  \n",
       "110      kd224        71     329  \n",
       "111      kd233        23     285  \n",
       "112      kd273        23     285  \n",
       "113      kd243        23     277  \n",
       "\n",
       "[114 rows x 13 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions = msa_regions_pdbs[['uniprot', 'term_id_cre', 'start_cre', 'end_cre', 'length_cre', 'term_id_kd', 'start_kd', 'end_kd']].drop_duplicates()\n",
    "pfam_cre = pfam.merge(regions)\n",
    "pfam_cre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Find wich CREs overlaps PFAM domains using Pyranges join method\n",
    "regions_pr = pr.PyRanges(regions.rename(columns={'uniprot': 'Chromosome', 'start_cre': 'Start', 'end_cre': 'End'}))\n",
    "pfam_pr = pr.PyRanges(pfam.rename(columns={'uniprot': 'Chromosome', 'start': 'Start', 'end': 'End'}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pfam</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>domain_name</th>\n",
       "      <th>term_id_cre</th>\n",
       "      <th>start_cre</th>\n",
       "      <th>end_cre</th>\n",
       "      <th>length_cre</th>\n",
       "      <th>term_id_kd</th>\n",
       "      <th>start_kd</th>\n",
       "      <th>end_kd</th>\n",
       "      <th>overlap</th>\n",
       "      <th>fraction_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O74536</td>\n",
       "      <td>PF08587</td>\n",
       "      <td>SNF1-like protein kinase ssp2</td>\n",
       "      <td>305</td>\n",
       "      <td>345</td>\n",
       "      <td>UBA_2</td>\n",
       "      <td>cre21</td>\n",
       "      <td>305</td>\n",
       "      <td>351</td>\n",
       "      <td>47</td>\n",
       "      <td>kd21</td>\n",
       "      <td>34</td>\n",
       "      <td>285</td>\n",
       "      <td>41</td>\n",
       "      <td>0.87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O96013</td>\n",
       "      <td>PF00786</td>\n",
       "      <td>Serine/threonine-protein kinase PAK 4</td>\n",
       "      <td>10</td>\n",
       "      <td>63</td>\n",
       "      <td>PBD</td>\n",
       "      <td>cre32</td>\n",
       "      <td>1</td>\n",
       "      <td>130</td>\n",
       "      <td>130</td>\n",
       "      <td>kd34</td>\n",
       "      <td>321</td>\n",
       "      <td>572</td>\n",
       "      <td>55</td>\n",
       "      <td>0.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P00519</td>\n",
       "      <td>PF00018</td>\n",
       "      <td>Tyrosine-protein kinase ABL1</td>\n",
       "      <td>67</td>\n",
       "      <td>113</td>\n",
       "      <td>SH3_1</td>\n",
       "      <td>cre36</td>\n",
       "      <td>61</td>\n",
       "      <td>233</td>\n",
       "      <td>173</td>\n",
       "      <td>kd37</td>\n",
       "      <td>242</td>\n",
       "      <td>493</td>\n",
       "      <td>48</td>\n",
       "      <td>0.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P00519</td>\n",
       "      <td>PF00017</td>\n",
       "      <td>Tyrosine-protein kinase ABL1</td>\n",
       "      <td>127</td>\n",
       "      <td>202</td>\n",
       "      <td>SH2</td>\n",
       "      <td>cre36</td>\n",
       "      <td>61</td>\n",
       "      <td>233</td>\n",
       "      <td>173</td>\n",
       "      <td>kd37</td>\n",
       "      <td>242</td>\n",
       "      <td>493</td>\n",
       "      <td>77</td>\n",
       "      <td>0.45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P00523</td>\n",
       "      <td>PF00018</td>\n",
       "      <td>Proto-oncogene tyrosine-protein kinase Src</td>\n",
       "      <td>87</td>\n",
       "      <td>134</td>\n",
       "      <td>SH3_1</td>\n",
       "      <td>cre37</td>\n",
       "      <td>88</td>\n",
       "      <td>248</td>\n",
       "      <td>161</td>\n",
       "      <td>kd38</td>\n",
       "      <td>267</td>\n",
       "      <td>520</td>\n",
       "      <td>47</td>\n",
       "      <td>0.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P00523</td>\n",
       "      <td>PF00017</td>\n",
       "      <td>Proto-oncogene tyrosine-protein kinase Src</td>\n",
       "      <td>148</td>\n",
       "      <td>230</td>\n",
       "      <td>SH2</td>\n",
       "      <td>cre37</td>\n",
       "      <td>88</td>\n",
       "      <td>248</td>\n",
       "      <td>161</td>\n",
       "      <td>kd38</td>\n",
       "      <td>267</td>\n",
       "      <td>520</td>\n",
       "      <td>84</td>\n",
       "      <td>0.52</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P00533</td>\n",
       "      <td>PF07714</td>\n",
       "      <td>Epidermal growth factor receptor</td>\n",
       "      <td>713</td>\n",
       "      <td>965</td>\n",
       "      <td>PK_Tyr_Ser</td>\n",
       "      <td>cre39</td>\n",
       "      <td>965</td>\n",
       "      <td>998</td>\n",
       "      <td>34</td>\n",
       "      <td>kd39</td>\n",
       "      <td>712</td>\n",
       "      <td>979</td>\n",
       "      <td>1</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P10721</td>\n",
       "      <td>PF07714</td>\n",
       "      <td>Mast/stem cell growth factor receptor Kit</td>\n",
       "      <td>589</td>\n",
       "      <td>923</td>\n",
       "      <td>PK_Tyr_Ser</td>\n",
       "      <td>cre51</td>\n",
       "      <td>544</td>\n",
       "      <td>591</td>\n",
       "      <td>48</td>\n",
       "      <td>kd51</td>\n",
       "      <td>589</td>\n",
       "      <td>937</td>\n",
       "      <td>3</td>\n",
       "      <td>0.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P28482</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Mitogen-activated protein kinase 1</td>\n",
       "      <td>25</td>\n",
       "      <td>313</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre66</td>\n",
       "      <td>173</td>\n",
       "      <td>196</td>\n",
       "      <td>24</td>\n",
       "      <td>kd66</td>\n",
       "      <td>25</td>\n",
       "      <td>313</td>\n",
       "      <td>23</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P29323</td>\n",
       "      <td>PF14575</td>\n",
       "      <td>Ephrin type-B receptor 2</td>\n",
       "      <td>544</td>\n",
       "      <td>617</td>\n",
       "      <td>EphA2_TM</td>\n",
       "      <td>cre71</td>\n",
       "      <td>588</td>\n",
       "      <td>621</td>\n",
       "      <td>34</td>\n",
       "      <td>kd71</td>\n",
       "      <td>621</td>\n",
       "      <td>884</td>\n",
       "      <td>30</td>\n",
       "      <td>0.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>P31749</td>\n",
       "      <td>PF00169</td>\n",
       "      <td>RAC-alpha serine/threonine-protein kinase</td>\n",
       "      <td>7</td>\n",
       "      <td>106</td>\n",
       "      <td>PH</td>\n",
       "      <td>cre72</td>\n",
       "      <td>17</td>\n",
       "      <td>149</td>\n",
       "      <td>133</td>\n",
       "      <td>kd72</td>\n",
       "      <td>150</td>\n",
       "      <td>408</td>\n",
       "      <td>90</td>\n",
       "      <td>0.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>P31751</td>\n",
       "      <td>PF00169</td>\n",
       "      <td>RAC-beta serine/threonine-protein kinase</td>\n",
       "      <td>6</td>\n",
       "      <td>106</td>\n",
       "      <td>PH</td>\n",
       "      <td>cre73</td>\n",
       "      <td>17</td>\n",
       "      <td>151</td>\n",
       "      <td>135</td>\n",
       "      <td>kd73</td>\n",
       "      <td>152</td>\n",
       "      <td>409</td>\n",
       "      <td>90</td>\n",
       "      <td>0.67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>P36507</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Dual specificity mitogen-activated protein kin...</td>\n",
       "      <td>72</td>\n",
       "      <td>369</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre78</td>\n",
       "      <td>75</td>\n",
       "      <td>102</td>\n",
       "      <td>28</td>\n",
       "      <td>kd78</td>\n",
       "      <td>72</td>\n",
       "      <td>369</td>\n",
       "      <td>27</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P45985</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Dual specificity mitogen-activated protein kin...</td>\n",
       "      <td>104</td>\n",
       "      <td>367</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre85</td>\n",
       "      <td>109</td>\n",
       "      <td>136</td>\n",
       "      <td>28</td>\n",
       "      <td>kd84</td>\n",
       "      <td>102</td>\n",
       "      <td>367</td>\n",
       "      <td>27</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Q8IU85</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>23</td>\n",
       "      <td>279</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre226</td>\n",
       "      <td>279</td>\n",
       "      <td>319</td>\n",
       "      <td>41</td>\n",
       "      <td>kd226</td>\n",
       "      <td>23</td>\n",
       "      <td>279</td>\n",
       "      <td>1</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Q96NX5</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>25</td>\n",
       "      <td>277</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre243</td>\n",
       "      <td>277</td>\n",
       "      <td>317</td>\n",
       "      <td>41</td>\n",
       "      <td>kd243</td>\n",
       "      <td>23</td>\n",
       "      <td>277</td>\n",
       "      <td>1</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Q00944</td>\n",
       "      <td>PF00373</td>\n",
       "      <td>Focal adhesion kinase 1</td>\n",
       "      <td>135</td>\n",
       "      <td>250</td>\n",
       "      <td>FERM_M</td>\n",
       "      <td>cre110</td>\n",
       "      <td>35</td>\n",
       "      <td>362</td>\n",
       "      <td>328</td>\n",
       "      <td>kd110</td>\n",
       "      <td>422</td>\n",
       "      <td>680</td>\n",
       "      <td>117</td>\n",
       "      <td>0.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Q00944</td>\n",
       "      <td>PF18038</td>\n",
       "      <td>Focal adhesion kinase 1</td>\n",
       "      <td>35</td>\n",
       "      <td>130</td>\n",
       "      <td>FERM_N_2</td>\n",
       "      <td>cre110</td>\n",
       "      <td>35</td>\n",
       "      <td>362</td>\n",
       "      <td>328</td>\n",
       "      <td>kd110</td>\n",
       "      <td>422</td>\n",
       "      <td>680</td>\n",
       "      <td>96</td>\n",
       "      <td>0.29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Q01974</td>\n",
       "      <td>PF07714</td>\n",
       "      <td>Tyrosine-protein kinase transmembrane receptor...</td>\n",
       "      <td>474</td>\n",
       "      <td>746</td>\n",
       "      <td>PK_Tyr_Ser</td>\n",
       "      <td>cre111</td>\n",
       "      <td>633</td>\n",
       "      <td>646</td>\n",
       "      <td>14</td>\n",
       "      <td>kd111</td>\n",
       "      <td>473</td>\n",
       "      <td>746</td>\n",
       "      <td>13</td>\n",
       "      <td>0.93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Q02750</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Dual specificity mitogen-activated protein kin...</td>\n",
       "      <td>68</td>\n",
       "      <td>361</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre112</td>\n",
       "      <td>75</td>\n",
       "      <td>102</td>\n",
       "      <td>28</td>\n",
       "      <td>kd112</td>\n",
       "      <td>68</td>\n",
       "      <td>361</td>\n",
       "      <td>27</td>\n",
       "      <td>0.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Q13153</td>\n",
       "      <td>PF00786</td>\n",
       "      <td>Serine/threonine-protein kinase PAK 1</td>\n",
       "      <td>74</td>\n",
       "      <td>132</td>\n",
       "      <td>PBD</td>\n",
       "      <td>cre123</td>\n",
       "      <td>83</td>\n",
       "      <td>149</td>\n",
       "      <td>67</td>\n",
       "      <td>kd123</td>\n",
       "      <td>270</td>\n",
       "      <td>521</td>\n",
       "      <td>50</td>\n",
       "      <td>0.75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Q14012</td>\n",
       "      <td>PF00069</td>\n",
       "      <td>Calcium/calmodulin-dependent protein kinase ty...</td>\n",
       "      <td>22</td>\n",
       "      <td>276</td>\n",
       "      <td>Pkinase</td>\n",
       "      <td>cre135</td>\n",
       "      <td>276</td>\n",
       "      <td>316</td>\n",
       "      <td>41</td>\n",
       "      <td>kd133</td>\n",
       "      <td>20</td>\n",
       "      <td>276</td>\n",
       "      <td>1</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Q14680</td>\n",
       "      <td>PF02149</td>\n",
       "      <td>Maternal embryonic leucine zipper kinase</td>\n",
       "      <td>608</td>\n",
       "      <td>650</td>\n",
       "      <td>KA1</td>\n",
       "      <td>cre136</td>\n",
       "      <td>326</td>\n",
       "      <td>651</td>\n",
       "      <td>326</td>\n",
       "      <td>kd134</td>\n",
       "      <td>11</td>\n",
       "      <td>263</td>\n",
       "      <td>44</td>\n",
       "      <td>0.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Q61846</td>\n",
       "      <td>PF02149</td>\n",
       "      <td>Maternal embryonic leucine zipper kinase</td>\n",
       "      <td>600</td>\n",
       "      <td>642</td>\n",
       "      <td>KA1</td>\n",
       "      <td>cre178</td>\n",
       "      <td>326</td>\n",
       "      <td>643</td>\n",
       "      <td>318</td>\n",
       "      <td>kd178</td>\n",
       "      <td>11</td>\n",
       "      <td>263</td>\n",
       "      <td>44</td>\n",
       "      <td>0.14</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   uniprot     pfam                                       protein_name  start  \\\n",
       "0   O74536  PF08587                      SNF1-like protein kinase ssp2    305   \n",
       "1   O96013  PF00786              Serine/threonine-protein kinase PAK 4     10   \n",
       "2   P00519  PF00018                       Tyrosine-protein kinase ABL1     67   \n",
       "3   P00519  PF00017                       Tyrosine-protein kinase ABL1    127   \n",
       "4   P00523  PF00018         Proto-oncogene tyrosine-protein kinase Src     87   \n",
       "5   P00523  PF00017         Proto-oncogene tyrosine-protein kinase Src    148   \n",
       "6   P00533  PF07714                   Epidermal growth factor receptor    713   \n",
       "7   P10721  PF07714          Mast/stem cell growth factor receptor Kit    589   \n",
       "8   P28482  PF00069                 Mitogen-activated protein kinase 1     25   \n",
       "9   P29323  PF14575                           Ephrin type-B receptor 2    544   \n",
       "10  P31749  PF00169          RAC-alpha serine/threonine-protein kinase      7   \n",
       "11  P31751  PF00169           RAC-beta serine/threonine-protein kinase      6   \n",
       "12  P36507  PF00069  Dual specificity mitogen-activated protein kin...     72   \n",
       "13  P45985  PF00069  Dual specificity mitogen-activated protein kin...    104   \n",
       "14  Q8IU85  PF00069  Calcium/calmodulin-dependent protein kinase ty...     23   \n",
       "15  Q96NX5  PF00069  Calcium/calmodulin-dependent protein kinase ty...     25   \n",
       "16  Q00944  PF00373                            Focal adhesion kinase 1    135   \n",
       "17  Q00944  PF18038                            Focal adhesion kinase 1     35   \n",
       "18  Q01974  PF07714  Tyrosine-protein kinase transmembrane receptor...    474   \n",
       "19  Q02750  PF00069  Dual specificity mitogen-activated protein kin...     68   \n",
       "20  Q13153  PF00786              Serine/threonine-protein kinase PAK 1     74   \n",
       "21  Q14012  PF00069  Calcium/calmodulin-dependent protein kinase ty...     22   \n",
       "22  Q14680  PF02149           Maternal embryonic leucine zipper kinase    608   \n",
       "23  Q61846  PF02149           Maternal embryonic leucine zipper kinase    600   \n",
       "\n",
       "    end domain_name term_id_cre  start_cre  end_cre  length_cre term_id_kd  \\\n",
       "0   345       UBA_2       cre21        305      351          47       kd21   \n",
       "1    63         PBD       cre32          1      130         130       kd34   \n",
       "2   113       SH3_1       cre36         61      233         173       kd37   \n",
       "3   202         SH2       cre36         61      233         173       kd37   \n",
       "4   134       SH3_1       cre37         88      248         161       kd38   \n",
       "5   230         SH2       cre37         88      248         161       kd38   \n",
       "6   965  PK_Tyr_Ser       cre39        965      998          34       kd39   \n",
       "7   923  PK_Tyr_Ser       cre51        544      591          48       kd51   \n",
       "8   313     Pkinase       cre66        173      196          24       kd66   \n",
       "9   617    EphA2_TM       cre71        588      621          34       kd71   \n",
       "10  106          PH       cre72         17      149         133       kd72   \n",
       "11  106          PH       cre73         17      151         135       kd73   \n",
       "12  369     Pkinase       cre78         75      102          28       kd78   \n",
       "13  367     Pkinase       cre85        109      136          28       kd84   \n",
       "14  279     Pkinase      cre226        279      319          41      kd226   \n",
       "15  277     Pkinase      cre243        277      317          41      kd243   \n",
       "16  250      FERM_M      cre110         35      362         328      kd110   \n",
       "17  130    FERM_N_2      cre110         35      362         328      kd110   \n",
       "18  746  PK_Tyr_Ser      cre111        633      646          14      kd111   \n",
       "19  361     Pkinase      cre112         75      102          28      kd112   \n",
       "20  132         PBD      cre123         83      149          67      kd123   \n",
       "21  276     Pkinase      cre135        276      316          41      kd133   \n",
       "22  650         KA1      cre136        326      651         326      kd134   \n",
       "23  642         KA1      cre178        326      643         318      kd178   \n",
       "\n",
       "    start_kd  end_kd  overlap  fraction_overlap  \n",
       "0         34     285       41              0.87  \n",
       "1        321     572       55              0.42  \n",
       "2        242     493       48              0.28  \n",
       "3        242     493       77              0.45  \n",
       "4        267     520       47              0.29  \n",
       "5        267     520       84              0.52  \n",
       "6        712     979        1              0.03  \n",
       "7        589     937        3              0.06  \n",
       "8         25     313       23              0.96  \n",
       "9        621     884       30              0.88  \n",
       "10       150     408       90              0.68  \n",
       "11       152     409       90              0.67  \n",
       "12        72     369       27              0.96  \n",
       "13       102     367       27              0.96  \n",
       "14        23     279        1              0.02  \n",
       "15        23     277        1              0.02  \n",
       "16       422     680      117              0.36  \n",
       "17       422     680       96              0.29  \n",
       "18       473     746       13              0.93  \n",
       "19        68     361       27              0.96  \n",
       "20       270     521       50              0.75  \n",
       "21        20     276        1              0.02  \n",
       "22        11     263       44              0.13  \n",
       "23        11     263       44              0.14  "
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapped = pfam_pr.join(regions_pr, slack= 1, suffix= '_cre', report_overlap= True)\n",
    "overlapped = overlapped.as_df()\n",
    "overlapped = overlapped.rename(columns= {'Chromosome': 'uniprot'})\n",
    "overlapped.columns = overlapped.columns.str.lower()\n",
    "overlapped[\"fraction_overlap\"] = round(overlapped.overlap / overlapped.length_cre, 2)\n",
    "overlapped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "#overlapped.to_csv('../datasets/pfam_cre_overlap.tsv', sep= '\\t', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Pkinase       7\n",
       "PK_Tyr_Ser    3\n",
       "KA1           2\n",
       "PBD           2\n",
       "SH3_1         2\n",
       "SH2           2\n",
       "PH            2\n",
       "FERM_N_2      1\n",
       "EphA2_TM      1\n",
       "UBA_2         1\n",
       "FERM_M        1\n",
       "Name: domain_name, dtype: int64"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapped.domain_name.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapped.term_id_cre.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprot</th>\n",
       "      <th>pfam</th>\n",
       "      <th>protein_name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>domain_name</th>\n",
       "      <th>term_id_cre</th>\n",
       "      <th>start_cre</th>\n",
       "      <th>end_cre</th>\n",
       "      <th>length_cre</th>\n",
       "      <th>term_id_kd</th>\n",
       "      <th>start_kd</th>\n",
       "      <th>end_kd</th>\n",
       "      <th>overlap</th>\n",
       "      <th>fraction_overlap</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P00519</td>\n",
       "      <td>PF00018</td>\n",
       "      <td>Tyrosine-protein kinase ABL1</td>\n",
       "      <td>67</td>\n",
       "      <td>113</td>\n",
       "      <td>SH3_1</td>\n",
       "      <td>cre36</td>\n",
       "      <td>61</td>\n",
       "      <td>233</td>\n",
       "      <td>173</td>\n",
       "      <td>kd37</td>\n",
       "      <td>242</td>\n",
       "      <td>493</td>\n",
       "      <td>48</td>\n",
       "      <td>0.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P00519</td>\n",
       "      <td>PF00017</td>\n",
       "      <td>Tyrosine-protein kinase ABL1</td>\n",
       "      <td>127</td>\n",
       "      <td>202</td>\n",
       "      <td>SH2</td>\n",
       "      <td>cre36</td>\n",
       "      <td>61</td>\n",
       "      <td>233</td>\n",
       "      <td>173</td>\n",
       "      <td>kd37</td>\n",
       "      <td>242</td>\n",
       "      <td>493</td>\n",
       "      <td>77</td>\n",
       "      <td>0.45</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  uniprot     pfam                  protein_name  start  end domain_name  \\\n",
       "2  P00519  PF00018  Tyrosine-protein kinase ABL1     67  113       SH3_1   \n",
       "3  P00519  PF00017  Tyrosine-protein kinase ABL1    127  202         SH2   \n",
       "\n",
       "  term_id_cre  start_cre  end_cre  length_cre term_id_kd  start_kd  end_kd  \\\n",
       "2       cre36         61      233         173       kd37       242     493   \n",
       "3       cre36         61      233         173       kd37       242     493   \n",
       "\n",
       "   overlap  fraction_overlap  \n",
       "2       48              0.28  \n",
       "3       77              0.45  "
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapped[overlapped.uniprot == 'P00519']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PFAM - CRE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.read_csv('../datasets/pfam_cre_overlap.tsv')"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Secondary structure scores - Fess scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Dataset with all proteins\n",
    "#msa_regions_pdbs = pd.read_csv('../datasets/msa_regions_pdbs.tsv', sep= '\\t')\n",
    "all_msas = pd.read_csv('../datasets/all_msas.tsv', sep= '\\t')\n",
    "all_msas = all_msas[all_msas.msa.str.endswith(\"_60\")]\n",
    "\n",
    "region_to_transfer = pd.read_csv('../homology_transfer/pipeline_MSA_uniprot/data_snake/12_region_to_transfer.txt', sep= \" \")\n",
    "region_to_transfer = region_to_transfer[(region_to_transfer.full_identity == '60') & (region_to_transfer.region_identity == '60')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "set1 = set(region_to_transfer.uniprot_acc.unique()) # unique proteins in our orthologs dataset\n",
    "set2 = set(all_msas.uniprot.unique())\n",
    "all = set1.union(set2)\n",
    "all = list(all)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "grouped_strings = []\n",
    "for i in range(0, len(all), 20):\n",
    "    group = all[i:i+20]\n",
    "    group_string = \",\".join(group)\n",
    "    grouped_strings.append(group_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chunk 1 downloaded and parsed. No. of entries 20\n",
      "chunk 2 downloaded and parsed. No. of entries 40\n",
      "chunk 3 downloaded and parsed. No. of entries 60\n",
      "chunk 4 downloaded and parsed. No. of entries 80\n",
      "chunk 5 downloaded and parsed. No. of entries 100\n",
      "chunk 6 downloaded and parsed. No. of entries 120\n",
      "chunk 7 downloaded and parsed. No. of entries 140\n",
      "chunk 8 downloaded and parsed. No. of entries 160\n",
      "chunk 9 downloaded and parsed. No. of entries 180\n",
      "chunk 10 downloaded and parsed. No. of entries 200\n",
      "chunk 11 downloaded and parsed. No. of entries 220\n",
      "chunk 12 downloaded and parsed. No. of entries 240\n",
      "chunk 13 downloaded and parsed. No. of entries 260\n",
      "chunk 14 downloaded and parsed. No. of entries 280\n",
      "chunk 15 downloaded and parsed. No. of entries 300\n",
      "chunk 16 downloaded and parsed. No. of entries 320\n",
      "chunk 17 downloaded and parsed. No. of entries 340\n",
      "chunk 18 downloaded and parsed. No. of entries 360\n",
      "chunk 19 downloaded and parsed. No. of entries 380\n",
      "chunk 20 downloaded and parsed. No. of entries 400\n",
      "chunk 21 downloaded and parsed. No. of entries 420\n",
      "chunk 22 downloaded and parsed. No. of entries 440\n",
      "chunk 23 downloaded and parsed. No. of entries 460\n",
      "chunk 24 downloaded and parsed. No. of entries 480\n",
      "chunk 25 downloaded and parsed. No. of entries 500\n",
      "chunk 26 downloaded and parsed. No. of entries 520\n",
      "chunk 27 downloaded and parsed. No. of entries 540\n",
      "chunk 28 downloaded and parsed. No. of entries 560\n",
      "chunk 29 downloaded and parsed. No. of entries 580\n",
      "chunk 30 downloaded and parsed. No. of entries 600\n",
      "chunk 31 downloaded and parsed. No. of entries 620\n",
      "chunk 32 downloaded and parsed. No. of entries 640\n",
      "chunk 33 downloaded and parsed. No. of entries 660\n",
      "chunk 34 downloaded and parsed. No. of entries 680\n",
      "chunk 35 downloaded and parsed. No. of entries 700\n",
      "chunk 36 downloaded and parsed. No. of entries 720\n",
      "chunk 37 downloaded and parsed. No. of entries 740\n",
      "chunk 38 downloaded and parsed. No. of entries 760\n",
      "chunk 39 downloaded and parsed. No. of entries 780\n",
      "chunk 40 downloaded and parsed. No. of entries 800\n",
      "chunk 41 downloaded and parsed. No. of entries 820\n",
      "chunk 42 downloaded and parsed. No. of entries 840\n",
      "chunk 43 downloaded and parsed. No. of entries 860\n",
      "chunk 44 downloaded and parsed. No. of entries 880\n",
      "chunk 45 downloaded and parsed. No. of entries 900\n",
      "chunk 46 downloaded and parsed. No. of entries 920\n",
      "chunk 47 downloaded and parsed. No. of entries 940\n",
      "chunk 48 downloaded and parsed. No. of entries 960\n",
      "chunk 49 downloaded and parsed. No. of entries 980\n",
      "chunk 50 downloaded and parsed. No. of entries 1000\n",
      "chunk 51 downloaded and parsed. No. of entries 1020\n",
      "chunk 52 downloaded and parsed. No. of entries 1040\n",
      "chunk 53 downloaded and parsed. No. of entries 1060\n",
      "chunk 54 downloaded and parsed. No. of entries 1080\n",
      "chunk 55 downloaded and parsed. No. of entries 1100\n",
      "chunk 56 downloaded and parsed. No. of entries 1120\n",
      "chunk 57 downloaded and parsed. No. of entries 1140\n",
      "chunk 58 downloaded and parsed. No. of entries 1160\n",
      "chunk 59 downloaded and parsed. No. of entries 1180\n",
      "chunk 60 downloaded and parsed. No. of entries 1200\n",
      "chunk 61 downloaded and parsed. No. of entries 1220\n",
      "chunk 62 downloaded and parsed. No. of entries 1240\n",
      "chunk 63 downloaded and parsed. No. of entries 1260\n",
      "chunk 64 downloaded and parsed. No. of entries 1280\n",
      "chunk 65 downloaded and parsed. No. of entries 1300\n",
      "chunk 66 downloaded and parsed. No. of entries 1320\n",
      "chunk 67 downloaded and parsed. No. of entries 1340\n",
      "chunk 68 downloaded and parsed. No. of entries 1360\n",
      "chunk 69 downloaded and parsed. No. of entries 1380\n",
      "chunk 70 downloaded and parsed. No. of entries 1400\n",
      "chunk 71 downloaded and parsed. No. of entries 1420\n",
      "chunk 72 downloaded and parsed. No. of entries 1440\n",
      "chunk 73 downloaded and parsed. No. of entries 1460\n",
      "chunk 74 downloaded and parsed. No. of entries 1480\n",
      "chunk 75 downloaded and parsed. No. of entries 1500\n",
      "chunk 76 downloaded and parsed. No. of entries 1520\n",
      "chunk 77 downloaded and parsed. No. of entries 1540\n",
      "chunk 78 downloaded and parsed. No. of entries 1560\n",
      "chunk 79 downloaded and parsed. No. of entries 1580\n",
      "chunk 80 downloaded and parsed. No. of entries 1600\n",
      "chunk 81 downloaded and parsed. No. of entries 1620\n",
      "chunk 82 downloaded and parsed. No. of entries 1640\n",
      "chunk 83 downloaded and parsed. No. of entries 1660\n",
      "chunk 84 downloaded and parsed. No. of entries 1680\n",
      "chunk 85 downloaded and parsed. No. of entries 1700\n",
      "chunk 86 downloaded and parsed. No. of entries 1720\n",
      "chunk 87 downloaded and parsed. No. of entries 1740\n",
      "chunk 88 downloaded and parsed. No. of entries 1760\n",
      "chunk 89 downloaded and parsed. No. of entries 1780\n",
      "chunk 90 downloaded and parsed. No. of entries 1800\n",
      "chunk 91 downloaded and parsed. No. of entries 1820\n",
      "chunk 92 downloaded and parsed. No. of entries 1840\n",
      "chunk 93 downloaded and parsed. No. of entries 1860\n",
      "chunk 94 downloaded and parsed. No. of entries 1880\n",
      "chunk 95 downloaded and parsed. No. of entries 1900\n",
      "chunk 96 downloaded and parsed. No. of entries 1920\n",
      "chunk 97 downloaded and parsed. No. of entries 1940\n",
      "chunk 98 downloaded and parsed. No. of entries 1960\n",
      "chunk 99 downloaded and parsed. No. of entries 1980\n",
      "chunk 100 downloaded and parsed. No. of entries 2000\n",
      "chunk 101 downloaded and parsed. No. of entries 2020\n",
      "chunk 102 downloaded and parsed. No. of entries 2040\n",
      "chunk 103 downloaded and parsed. No. of entries 2060\n",
      "chunk 104 downloaded and parsed. No. of entries 2080\n",
      "chunk 105 downloaded and parsed. No. of entries 2100\n",
      "chunk 106 downloaded and parsed. No. of entries 2120\n",
      "chunk 107 downloaded and parsed. No. of entries 2140\n",
      "chunk 108 downloaded and parsed. No. of entries 2160\n",
      "chunk 109 downloaded and parsed. No. of entries 2180\n",
      "chunk 110 downloaded and parsed. No. of entries 2200\n",
      "chunk 111 downloaded and parsed. No. of entries 2220\n",
      "chunk 112 downloaded and parsed. No. of entries 2240\n",
      "chunk 113 downloaded and parsed. No. of entries 2260\n",
      "chunk 114 downloaded and parsed. No. of entries 2280\n",
      "chunk 115 downloaded and parsed. No. of entries 2300\n",
      "chunk 116 downloaded and parsed. No. of entries 2320\n",
      "chunk 117 downloaded and parsed. No. of entries 2340\n",
      "chunk 118 downloaded and parsed. No. of entries 2360\n",
      "chunk 119 downloaded and parsed. No. of entries 2380\n",
      "chunk 120 downloaded and parsed. No. of entries 2400\n",
      "chunk 121 downloaded and parsed. No. of entries 2420\n",
      "chunk 122 downloaded and parsed. No. of entries 2440\n",
      "chunk 123 downloaded and parsed. No. of entries 2460\n",
      "chunk 124 downloaded and parsed. No. of entries 2480\n",
      "chunk 125 downloaded and parsed. No. of entries 2500\n",
      "chunk 126 downloaded and parsed. No. of entries 2520\n",
      "chunk 127 downloaded and parsed. No. of entries 2540\n",
      "chunk 128 downloaded and parsed. No. of entries 2560\n",
      "chunk 129 downloaded and parsed. No. of entries 2580\n",
      "chunk 130 downloaded and parsed. No. of entries 2600\n",
      "chunk 131 downloaded and parsed. No. of entries 2620\n",
      "chunk 132 downloaded and parsed. No. of entries 2640\n",
      "chunk 133 downloaded and parsed. No. of entries 2660\n",
      "chunk 134 downloaded and parsed. No. of entries 2680\n",
      "chunk 135 downloaded and parsed. No. of entries 2700\n",
      "chunk 136 downloaded and parsed. No. of entries 2720\n",
      "chunk 137 downloaded and parsed. No. of entries 2740\n",
      "chunk 138 downloaded and parsed. No. of entries 2760\n",
      "chunk 139 downloaded and parsed. No. of entries 2780\n",
      "chunk 140 downloaded and parsed. No. of entries 2800\n",
      "chunk 141 downloaded and parsed. No. of entries 2820\n",
      "chunk 142 downloaded and parsed. No. of entries 2840\n",
      "chunk 143 downloaded and parsed. No. of entries 2860\n",
      "chunk 144 downloaded and parsed. No. of entries 2880\n",
      "chunk 145 downloaded and parsed. No. of entries 2900\n",
      "chunk 146 downloaded and parsed. No. of entries 2920\n",
      "chunk 147 downloaded and parsed. No. of entries 2940\n",
      "chunk 148 downloaded and parsed. No. of entries 2960\n",
      "chunk 149 downloaded and parsed. No. of entries 2980\n",
      "chunk 150 downloaded and parsed. No. of entries 3000\n",
      "chunk 151 downloaded and parsed. No. of entries 3020\n",
      "chunk 152 downloaded and parsed. No. of entries 3040\n",
      "chunk 153 downloaded and parsed. No. of entries 3060\n",
      "chunk 154 downloaded and parsed. No. of entries 3080\n",
      "chunk 155 downloaded and parsed. No. of entries 3100\n",
      "chunk 156 downloaded and parsed. No. of entries 3120\n",
      "chunk 157 downloaded and parsed. No. of entries 3140\n",
      "chunk 158 downloaded and parsed. No. of entries 3160\n",
      "chunk 159 downloaded and parsed. No. of entries 3180\n",
      "chunk 160 downloaded and parsed. No. of entries 3200\n",
      "chunk 161 downloaded and parsed. No. of entries 3220\n",
      "chunk 162 downloaded and parsed. No. of entries 3240\n",
      "chunk 163 downloaded and parsed. No. of entries 3260\n",
      "chunk 164 downloaded and parsed. No. of entries 3280\n",
      "chunk 165 downloaded and parsed. No. of entries 3300\n",
      "chunk 166 downloaded and parsed. No. of entries 3320\n",
      "chunk 167 downloaded and parsed. No. of entries 3340\n",
      "chunk 168 downloaded and parsed. No. of entries 3360\n",
      "chunk 169 downloaded and parsed. No. of entries 3380\n",
      "chunk 170 downloaded and parsed. No. of entries 3400\n",
      "chunk 171 downloaded and parsed. No. of entries 3420\n",
      "chunk 172 downloaded and parsed. No. of entries 3440\n",
      "chunk 173 downloaded and parsed. No. of entries 3460\n",
      "chunk 174 downloaded and parsed. No. of entries 3480\n",
      "chunk 175 downloaded and parsed. No. of entries 3500\n",
      "chunk 176 downloaded and parsed. No. of entries 3520\n",
      "chunk 177 downloaded and parsed. No. of entries 3540\n",
      "chunk 178 downloaded and parsed. No. of entries 3560\n",
      "chunk 179 downloaded and parsed. No. of entries 3580\n",
      "chunk 180 downloaded and parsed. No. of entries 3600\n",
      "chunk 181 downloaded and parsed. No. of entries 3620\n",
      "chunk 182 downloaded and parsed. No. of entries 3640\n",
      "chunk 183 downloaded and parsed. No. of entries 3660\n",
      "chunk 184 downloaded and parsed. No. of entries 3680\n",
      "chunk 185 downloaded and parsed. No. of entries 3700\n",
      "chunk 186 downloaded and parsed. No. of entries 3720\n",
      "chunk 187 downloaded and parsed. No. of entries 3740\n",
      "chunk 188 downloaded and parsed. No. of entries 3760\n",
      "chunk 189 downloaded and parsed. No. of entries 3780\n",
      "chunk 190 downloaded and parsed. No. of entries 3800\n",
      "chunk 191 downloaded and parsed. No. of entries 3820\n",
      "chunk 192 downloaded and parsed. No. of entries 3840\n",
      "chunk 193 downloaded and parsed. No. of entries 3860\n",
      "chunk 194 downloaded and parsed. No. of entries 3880\n",
      "chunk 195 downloaded and parsed. No. of entries 3900\n",
      "chunk 196 downloaded and parsed. No. of entries 3920\n",
      "chunk 197 downloaded and parsed. No. of entries 3940\n",
      "chunk 198 downloaded and parsed. No. of entries 3960\n",
      "chunk 199 downloaded and parsed. No. of entries 3980\n",
      "chunk 200 downloaded and parsed. No. of entries 4000\n",
      "chunk 201 downloaded and parsed. No. of entries 4020\n",
      "chunk 202 downloaded and parsed. No. of entries 4040\n",
      "chunk 203 downloaded and parsed. No. of entries 4060\n",
      "chunk 204 downloaded and parsed. No. of entries 4080\n",
      "chunk 205 downloaded and parsed. No. of entries 4100\n",
      "chunk 206 downloaded and parsed. No. of entries 4120\n",
      "chunk 207 downloaded and parsed. No. of entries 4140\n",
      "chunk 208 downloaded and parsed. No. of entries 4160\n",
      "chunk 209 downloaded and parsed. No. of entries 4180\n",
      "chunk 210 downloaded and parsed. No. of entries 4200\n",
      "chunk 211 downloaded and parsed. No. of entries 4220\n",
      "chunk 212 downloaded and parsed. No. of entries 4240\n",
      "chunk 213 downloaded and parsed. No. of entries 4260\n",
      "chunk 214 downloaded and parsed. No. of entries 4280\n",
      "chunk 215 downloaded and parsed. No. of entries 4300\n",
      "chunk 216 downloaded and parsed. No. of entries 4320\n",
      "chunk 217 downloaded and parsed. No. of entries 4340\n",
      "chunk 218 downloaded and parsed. No. of entries 4360\n",
      "chunk 219 downloaded and parsed. No. of entries 4380\n",
      "chunk 220 downloaded and parsed. No. of entries 4400\n",
      "chunk 221 downloaded and parsed. No. of entries 4420\n",
      "chunk 222 downloaded and parsed. No. of entries 4440\n",
      "chunk 223 downloaded and parsed. No. of entries 4460\n",
      "chunk 224 downloaded and parsed. No. of entries 4480\n",
      "chunk 225 downloaded and parsed. No. of entries 4500\n",
      "chunk 226 downloaded and parsed. No. of entries 4520\n",
      "chunk 227 downloaded and parsed. No. of entries 4540\n",
      "chunk 228 downloaded and parsed. No. of entries 4560\n",
      "chunk 229 downloaded and parsed. No. of entries 4580\n",
      "chunk 230 downloaded and parsed. No. of entries 4600\n",
      "chunk 231 downloaded and parsed. No. of entries 4620\n",
      "chunk 232 downloaded and parsed. No. of entries 4640\n",
      "chunk 233 downloaded and parsed. No. of entries 4660\n",
      "chunk 234 downloaded and parsed. No. of entries 4680\n",
      "chunk 235 downloaded and parsed. No. of entries 4700\n",
      "chunk 236 downloaded and parsed. No. of entries 4720\n",
      "chunk 237 downloaded and parsed. No. of entries 4740\n",
      "chunk 238 downloaded and parsed. No. of entries 4760\n",
      "chunk 239 downloaded and parsed. No. of entries 4780\n",
      "chunk 240 downloaded and parsed. No. of entries 4800\n",
      "chunk 241 downloaded and parsed. No. of entries 4820\n",
      "chunk 242 downloaded and parsed. No. of entries 4840\n",
      "chunk 243 downloaded and parsed. No. of entries 4860\n",
      "chunk 244 downloaded and parsed. No. of entries 4880\n",
      "chunk 245 downloaded and parsed. No. of entries 4900\n",
      "chunk 246 downloaded and parsed. No. of entries 4920\n",
      "chunk 247 downloaded and parsed. No. of entries 4940\n",
      "chunk 248 downloaded and parsed. No. of entries 4960\n",
      "chunk 249 downloaded and parsed. No. of entries 4980\n",
      "chunk 250 downloaded and parsed. No. of entries 5000\n",
      "chunk 251 downloaded and parsed. No. of entries 5020\n",
      "chunk 252 downloaded and parsed. No. of entries 5040\n",
      "chunk 253 downloaded and parsed. No. of entries 5060\n",
      "chunk 254 downloaded and parsed. No. of entries 5080\n",
      "chunk 255 downloaded and parsed. No. of entries 5100\n",
      "chunk 256 downloaded and parsed. No. of entries 5120\n",
      "chunk 257 downloaded and parsed. No. of entries 5140\n",
      "chunk 258 downloaded and parsed. No. of entries 5160\n",
      "chunk 259 downloaded and parsed. No. of entries 5180\n",
      "chunk 260 downloaded and parsed. No. of entries 5200\n",
      "chunk 261 downloaded and parsed. No. of entries 5220\n",
      "chunk 262 downloaded and parsed. No. of entries 5240\n",
      "chunk 263 downloaded and parsed. No. of entries 5260\n",
      "chunk 264 downloaded and parsed. No. of entries 5280\n",
      "chunk 265 downloaded and parsed. No. of entries 5300\n",
      "chunk 266 downloaded and parsed. No. of entries 5320\n",
      "chunk 267 downloaded and parsed. No. of entries 5340\n",
      "chunk 268 downloaded and parsed. No. of entries 5360\n",
      "chunk 269 downloaded and parsed. No. of entries 5380\n",
      "chunk 270 downloaded and parsed. No. of entries 5400\n",
      "chunk 271 downloaded and parsed. No. of entries 5420\n",
      "chunk 272 downloaded and parsed. No. of entries 5440\n",
      "chunk 273 downloaded and parsed. No. of entries 5460\n",
      "chunk 274 downloaded and parsed. No. of entries 5480\n",
      "chunk 275 downloaded and parsed. No. of entries 5500\n",
      "chunk 276 downloaded and parsed. No. of entries 5520\n",
      "chunk 277 downloaded and parsed. No. of entries 5540\n",
      "chunk 278 downloaded and parsed. No. of entries 5560\n",
      "chunk 279 downloaded and parsed. No. of entries 5580\n",
      "chunk 280 downloaded and parsed. No. of entries 5600\n",
      "chunk 281 downloaded and parsed. No. of entries 5620\n",
      "chunk 282 downloaded and parsed. No. of entries 5640\n",
      "chunk 283 downloaded and parsed. No. of entries 5660\n",
      "chunk 284 downloaded and parsed. No. of entries 5680\n",
      "chunk 285 downloaded and parsed. No. of entries 5700\n",
      "chunk 286 downloaded and parsed. No. of entries 5720\n",
      "chunk 287 downloaded and parsed. No. of entries 5740\n",
      "chunk 288 downloaded and parsed. No. of entries 5760\n",
      "chunk 289 downloaded and parsed. No. of entries 5780\n",
      "chunk 290 downloaded and parsed. No. of entries 5800\n",
      "chunk 291 downloaded and parsed. No. of entries 5820\n",
      "chunk 292 downloaded and parsed. No. of entries 5840\n",
      "chunk 293 downloaded and parsed. No. of entries 5860\n",
      "chunk 294 downloaded and parsed. No. of entries 5880\n",
      "chunk 295 downloaded and parsed. No. of entries 5900\n",
      "chunk 296 downloaded and parsed. No. of entries 5920\n",
      "chunk 297 downloaded and parsed. No. of entries 5940\n",
      "chunk 298 downloaded and parsed. No. of entries 5960\n",
      "chunk 299 downloaded and parsed. No. of entries 5980\n",
      "chunk 300 downloaded and parsed. No. of entries 6000\n",
      "chunk 301 downloaded and parsed. No. of entries 6020\n",
      "chunk 302 downloaded and parsed. No. of entries 6040\n",
      "chunk 303 downloaded and parsed. No. of entries 6060\n",
      "chunk 304 downloaded and parsed. No. of entries 6080\n",
      "chunk 305 downloaded and parsed. No. of entries 6100\n",
      "chunk 306 downloaded and parsed. No. of entries 6120\n",
      "chunk 307 downloaded and parsed. No. of entries 6140\n",
      "chunk 308 downloaded and parsed. No. of entries 6160\n",
      "chunk 309 downloaded and parsed. No. of entries 6180\n",
      "chunk 310 downloaded and parsed. No. of entries 6200\n",
      "chunk 311 downloaded and parsed. No. of entries 6220\n",
      "chunk 312 downloaded and parsed. No. of entries 6240\n",
      "chunk 313 downloaded and parsed. No. of entries 6260\n",
      "chunk 314 downloaded and parsed. No. of entries 6280\n",
      "chunk 315 downloaded and parsed. No. of entries 6300\n",
      "chunk 316 downloaded and parsed. No. of entries 6320\n",
      "chunk 317 downloaded and parsed. No. of entries 6340\n",
      "chunk 318 downloaded and parsed. No. of entries 6360\n",
      "chunk 319 downloaded and parsed. No. of entries 6380\n",
      "chunk 320 downloaded and parsed. No. of entries 6400\n",
      "chunk 321 downloaded and parsed. No. of entries 6420\n",
      "chunk 322 downloaded and parsed. No. of entries 6440\n",
      "chunk 323 downloaded and parsed. No. of entries 6460\n",
      "chunk 324 downloaded and parsed. No. of entries 6480\n",
      "chunk 325 downloaded and parsed. No. of entries 6500\n",
      "chunk 326 downloaded and parsed. No. of entries 6520\n",
      "chunk 327 downloaded and parsed. No. of entries 6540\n",
      "chunk 328 downloaded and parsed. No. of entries 6560\n",
      "chunk 329 downloaded and parsed. No. of entries 6580\n",
      "chunk 330 downloaded and parsed. No. of entries 6600\n",
      "chunk 331 downloaded and parsed. No. of entries 6620\n",
      "chunk 332 downloaded and parsed. No. of entries 6640\n",
      "chunk 333 downloaded and parsed. No. of entries 6660\n",
      "chunk 334 downloaded and parsed. No. of entries 6680\n",
      "chunk 335 downloaded and parsed. No. of entries 6700\n",
      "chunk 336 downloaded and parsed. No. of entries 6720\n",
      "chunk 337 downloaded and parsed. No. of entries 6740\n",
      "chunk 338 downloaded and parsed. No. of entries 6760\n",
      "chunk 339 downloaded and parsed. No. of entries 6780\n",
      "chunk 340 downloaded and parsed. No. of entries 6800\n",
      "chunk 341 downloaded and parsed. No. of entries 6820\n",
      "chunk 342 downloaded and parsed. No. of entries 6840\n",
      "chunk 343 downloaded and parsed. No. of entries 6860\n",
      "chunk 344 downloaded and parsed. No. of entries 6880\n",
      "chunk 345 downloaded and parsed. No. of entries 6900\n",
      "chunk 346 downloaded and parsed. No. of entries 6920\n",
      "chunk 347 downloaded and parsed. No. of entries 6940\n",
      "chunk 348 downloaded and parsed. No. of entries 6960\n",
      "chunk 349 downloaded and parsed. No. of entries 6980\n",
      "chunk 350 downloaded and parsed. No. of entries 7000\n",
      "chunk 351 downloaded and parsed. No. of entries 7020\n",
      "chunk 352 downloaded and parsed. No. of entries 7040\n",
      "chunk 353 downloaded and parsed. No. of entries 7060\n",
      "chunk 354 downloaded and parsed. No. of entries 7080\n",
      "chunk 355 downloaded and parsed. No. of entries 7100\n",
      "chunk 356 downloaded and parsed. No. of entries 7120\n",
      "chunk 357 downloaded and parsed. No. of entries 7140\n",
      "chunk 358 downloaded and parsed. No. of entries 7160\n",
      "chunk 359 downloaded and parsed. No. of entries 7180\n",
      "chunk 360 downloaded and parsed. No. of entries 7200\n",
      "chunk 361 downloaded and parsed. No. of entries 7220\n",
      "chunk 362 downloaded and parsed. No. of entries 7240\n",
      "chunk 363 downloaded and parsed. No. of entries 7260\n",
      "chunk 364 downloaded and parsed. No. of entries 7280\n",
      "chunk 365 downloaded and parsed. No. of entries 7300\n",
      "chunk 366 downloaded and parsed. No. of entries 7320\n",
      "chunk 367 downloaded and parsed. No. of entries 7340\n",
      "chunk 368 downloaded and parsed. No. of entries 7360\n",
      "chunk 369 downloaded and parsed. No. of entries 7380\n",
      "chunk 370 downloaded and parsed. No. of entries 7400\n",
      "chunk 371 downloaded and parsed. No. of entries 7420\n",
      "chunk 372 downloaded and parsed. No. of entries 7440\n",
      "chunk 373 downloaded and parsed. No. of entries 7460\n",
      "chunk 374 downloaded and parsed. No. of entries 7480\n",
      "chunk 375 downloaded and parsed. No. of entries 7500\n",
      "chunk 376 downloaded and parsed. No. of entries 7520\n",
      "chunk 377 downloaded and parsed. No. of entries 7540\n",
      "chunk 378 downloaded and parsed. No. of entries 7560\n",
      "chunk 379 downloaded and parsed. No. of entries 7580\n",
      "chunk 380 downloaded and parsed. No. of entries 7600\n",
      "chunk 381 downloaded and parsed. No. of entries 7620\n",
      "chunk 382 downloaded and parsed. No. of entries 7640\n",
      "chunk 383 downloaded and parsed. No. of entries 7660\n",
      "chunk 384 downloaded and parsed. No. of entries 7680\n",
      "chunk 385 downloaded and parsed. No. of entries 7700\n",
      "chunk 386 downloaded and parsed. No. of entries 7720\n",
      "chunk 387 downloaded and parsed. No. of entries 7740\n",
      "chunk 388 downloaded and parsed. No. of entries 7760\n",
      "chunk 389 downloaded and parsed. No. of entries 7780\n",
      "chunk 390 downloaded and parsed. No. of entries 7800\n",
      "chunk 391 downloaded and parsed. No. of entries 7820\n",
      "chunk 392 downloaded and parsed. No. of entries 7840\n",
      "chunk 393 downloaded and parsed. No. of entries 7860\n",
      "chunk 394 downloaded and parsed. No. of entries 7880\n",
      "chunk 395 downloaded and parsed. No. of entries 7900\n",
      "chunk 396 downloaded and parsed. No. of entries 7920\n",
      "chunk 397 downloaded and parsed. No. of entries 7940\n",
      "chunk 398 downloaded and parsed. No. of entries 7960\n",
      "chunk 399 downloaded and parsed. No. of entries 7980\n",
      "chunk 400 downloaded and parsed. No. of entries 8000\n",
      "chunk 401 downloaded and parsed. No. of entries 8020\n",
      "chunk 402 downloaded and parsed. No. of entries 8040\n",
      "chunk 403 downloaded and parsed. No. of entries 8060\n",
      "chunk 404 downloaded and parsed. No. of entries 8080\n",
      "chunk 405 downloaded and parsed. No. of entries 8100\n",
      "chunk 406 downloaded and parsed. No. of entries 8120\n",
      "chunk 407 downloaded and parsed. No. of entries 8140\n",
      "chunk 408 downloaded and parsed. No. of entries 8160\n",
      "chunk 409 downloaded and parsed. No. of entries 8180\n",
      "chunk 410 downloaded and parsed. No. of entries 8200\n",
      "chunk 411 downloaded and parsed. No. of entries 8220\n",
      "chunk 412 downloaded and parsed. No. of entries 8240\n",
      "chunk 413 downloaded and parsed. No. of entries 8260\n",
      "chunk 414 downloaded and parsed. No. of entries 8280\n",
      "chunk 415 downloaded and parsed. No. of entries 8300\n",
      "chunk 416 downloaded and parsed. No. of entries 8320\n",
      "chunk 417 downloaded and parsed. No. of entries 8340\n",
      "chunk 418 downloaded and parsed. No. of entries 8360\n",
      "chunk 419 downloaded and parsed. No. of entries 8380\n",
      "chunk 420 downloaded and parsed. No. of entries 8400\n",
      "chunk 421 downloaded and parsed. No. of entries 8411\n"
     ]
    }
   ],
   "source": [
    "# Empty df to concat the new data\n",
    "# cols to keep\n",
    "columns = ['acc', 'length', 'organism', 'ncbi_taxon_id', 'sequence',\n",
    "       'homology_domain_pfam', 'homology_domain_merge',\n",
    "       'prediction_disorder_mobidb_lite',  \"prediction_helix_fess\", \"prediction_sheet_fess\", \"prediction_coil_fess\"] # 'homology_disorder_disprot',\n",
    "\n",
    "mobidb_lite = pd.DataFrame() #columns= columns\n",
    "\n",
    "i= 0\n",
    "errors = {}\n",
    "# Empty list to store raw JSON data\n",
    "raw_json_data = []\n",
    "\n",
    "for chunk in grouped_strings: \n",
    "    # API download mobidb\n",
    "    to_download = f'https://mobidb.bio.unipd.it/api/download?acc={chunk}&projection='\n",
    "    # Request\n",
    "    response = requests.get(to_download)\n",
    "\n",
    "    if response.status_code == 200:\n",
    "        try:\n",
    "            # Store raw JSON data\n",
    "            raw_json_data.append(response.text)\n",
    "\n",
    "            # Replace blank spaces with a comma so json loads can read it properly\n",
    "            json_data = response.text.replace(\"}\\n{\", \"},{\")\n",
    "            j_text = \"[\" + json_data + \"]\"\n",
    "            # Parse the json string\n",
    "            query = json.loads(j_text)\n",
    "\n",
    "            # To dataframe\n",
    "            df = pd.DataFrame(query)\n",
    "            # Keep cols of interest\n",
    "            df.columns = df.columns.str.replace(\"-\", \"_\")\n",
    "            df = df[columns]\n",
    "            \n",
    "            # Add the new data\n",
    "            mobidb_lite = pd.concat([mobidb_lite, df])\n",
    "\n",
    "            i += 1\n",
    "            print(f\"chunk {i} downloaded and parsed. No. of entries {mobidb_lite.shape[0]}\")\n",
    "        \n",
    "        except BaseException as e:\n",
    "            print(f'an exception occured in chunk {i}: {e}')\n",
    "            continue\n",
    "\n",
    "    else:\n",
    "        errors[i] = to_download\n",
    "        print(f\"error with {to_download}\")\n",
    "\n",
    "# Concatenate raw JSON data and save to a file\n",
    "concatenated_json = \"[\" + \",\".join(raw_json_data) + \"]\"\n",
    "with open(\"../raw_data/mobidb_query_all_orthologs.json\", \"w\") as json_file:\n",
    "    json_file.write(concatenated_json)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mobidb_lite.to_csv('../raw_data/mobidb_query_all_orthologs.tsv', sep= '\\t', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>acc</th>\n",
       "      <th>length</th>\n",
       "      <th>prediction_helix_fess</th>\n",
       "      <th>prediction_sheet_fess</th>\n",
       "      <th>prediction_coil_fess</th>\n",
       "      <th>score_helix_fess</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A091DQI4</td>\n",
       "      <td>973</td>\n",
       "      <td>{'scores': [0.004, 0.068, 0.114, 0.153, 0.145,...</td>\n",
       "      <td>{'scores': [0.006, 0.454, 0.59, 0.617, 0.618, ...</td>\n",
       "      <td>{'scores': [0.99, 0.478, 0.296, 0.229, 0.237, ...</td>\n",
       "      <td>[0.004, 0.068, 0.114, 0.153, 0.145, 0.125, 0.1...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0A2V081</td>\n",
       "      <td>315</td>\n",
       "      <td>{'scores': [0.003, 0.028, 0.03, 0.011, 0.013, ...</td>\n",
       "      <td>{'scores': [0.002, 0.044, 0.051, 0.037, 0.016,...</td>\n",
       "      <td>{'scores': [0.995, 0.927, 0.919, 0.952, 0.97, ...</td>\n",
       "      <td>[0.003, 0.028, 0.03, 0.011, 0.013, 0.054, 0.12...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0D9R447</td>\n",
       "      <td>433</td>\n",
       "      <td>{'scores': [0.003, 0.021, 0.02, 0.018, 0.054, ...</td>\n",
       "      <td>{'scores': [0.002, 0.064, 0.068, 0.107, 0.141,...</td>\n",
       "      <td>{'scores': [0.995, 0.915, 0.912, 0.874, 0.805,...</td>\n",
       "      <td>[0.003, 0.021, 0.02, 0.018, 0.054, 0.082, 0.12...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D9RZF3</td>\n",
       "      <td>974</td>\n",
       "      <td>{'scores': [0.003, 0.018, 0.019, 0.023, 0.027,...</td>\n",
       "      <td>{'scores': [0.01, 0.516, 0.633, 0.636, 0.653, ...</td>\n",
       "      <td>{'scores': [0.987, 0.467, 0.348, 0.341, 0.32, ...</td>\n",
       "      <td>[0.003, 0.018, 0.019, 0.023, 0.027, 0.03, 0.03...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0E0IYS8</td>\n",
       "      <td>513</td>\n",
       "      <td>{'scores': [0.003, 0.011, 0.013, 0.01, 0.038, ...</td>\n",
       "      <td>{'scores': [0.002, 0.055, 0.064, 0.028, 0.027,...</td>\n",
       "      <td>{'scores': [0.995, 0.933, 0.923, 0.962, 0.935,...</td>\n",
       "      <td>[0.003, 0.011, 0.013, 0.01, 0.038, 0.036, 0.03...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8406</th>\n",
       "      <td>D6W6N7</td>\n",
       "      <td>359</td>\n",
       "      <td>{'scores': [0.003, 0.039, 0.119, 0.146, 0.157,...</td>\n",
       "      <td>{'scores': [0.002, 0.071, 0.197, 0.187, 0.134,...</td>\n",
       "      <td>{'scores': [0.995, 0.89, 0.684, 0.666, 0.709, ...</td>\n",
       "      <td>[0.003, 0.039, 0.119, 0.146, 0.157, 0.236, 0.2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8407</th>\n",
       "      <td>G5EA63</td>\n",
       "      <td>395</td>\n",
       "      <td>{'scores': [0.003, 0.015, 0.058, 0.035, 0.052,...</td>\n",
       "      <td>{'scores': [0.002, 0.034, 0.028, 0.023, 0.034,...</td>\n",
       "      <td>{'scores': [0.995, 0.951, 0.914, 0.942, 0.914,...</td>\n",
       "      <td>[0.003, 0.015, 0.058, 0.035, 0.052, 0.05, 0.06...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8408</th>\n",
       "      <td>H0Z3Y5</td>\n",
       "      <td>1130</td>\n",
       "      <td>{'scores': [0.003, 0.028, 0.041, 0.063, 0.08, ...</td>\n",
       "      <td>{'scores': [0.003, 0.363, 0.592, 0.684, 0.671,...</td>\n",
       "      <td>{'scores': [0.994, 0.609, 0.367, 0.253, 0.249,...</td>\n",
       "      <td>[0.003, 0.028, 0.041, 0.063, 0.08, 0.126, 0.11...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8409</th>\n",
       "      <td>P36507</td>\n",
       "      <td>400</td>\n",
       "      <td>{'scores': [0.004, 0.167, 0.198, 0.199, 0.121,...</td>\n",
       "      <td>{'scores': [0.002, 0.088, 0.092, 0.107, 0.118,...</td>\n",
       "      <td>{'scores': [0.994, 0.746, 0.711, 0.694, 0.76, ...</td>\n",
       "      <td>[0.004, 0.167, 0.198, 0.199, 0.121, 0.046, 0.0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8410</th>\n",
       "      <td>U3JUJ8</td>\n",
       "      <td>1150</td>\n",
       "      <td>{'scores': [0.003, 0.057, 0.21, 0.346, 0.464, ...</td>\n",
       "      <td>{'scores': [0.002, 0.054, 0.041, 0.032, 0.054,...</td>\n",
       "      <td>{'scores': [0.995, 0.889, 0.748, 0.622, 0.482,...</td>\n",
       "      <td>[0.003, 0.057, 0.21, 0.346, 0.464, 0.49, 0.637...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8411 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             acc  length                              prediction_helix_fess  \\\n",
       "0     A0A091DQI4     973  {'scores': [0.004, 0.068, 0.114, 0.153, 0.145,...   \n",
       "1     A0A0A2V081     315  {'scores': [0.003, 0.028, 0.03, 0.011, 0.013, ...   \n",
       "2     A0A0D9R447     433  {'scores': [0.003, 0.021, 0.02, 0.018, 0.054, ...   \n",
       "3     A0A0D9RZF3     974  {'scores': [0.003, 0.018, 0.019, 0.023, 0.027,...   \n",
       "4     A0A0E0IYS8     513  {'scores': [0.003, 0.011, 0.013, 0.01, 0.038, ...   \n",
       "...          ...     ...                                                ...   \n",
       "8406      D6W6N7     359  {'scores': [0.003, 0.039, 0.119, 0.146, 0.157,...   \n",
       "8407      G5EA63     395  {'scores': [0.003, 0.015, 0.058, 0.035, 0.052,...   \n",
       "8408      H0Z3Y5    1130  {'scores': [0.003, 0.028, 0.041, 0.063, 0.08, ...   \n",
       "8409      P36507     400  {'scores': [0.004, 0.167, 0.198, 0.199, 0.121,...   \n",
       "8410      U3JUJ8    1150  {'scores': [0.003, 0.057, 0.21, 0.346, 0.464, ...   \n",
       "\n",
       "                                  prediction_sheet_fess  \\\n",
       "0     {'scores': [0.006, 0.454, 0.59, 0.617, 0.618, ...   \n",
       "1     {'scores': [0.002, 0.044, 0.051, 0.037, 0.016,...   \n",
       "2     {'scores': [0.002, 0.064, 0.068, 0.107, 0.141,...   \n",
       "3     {'scores': [0.01, 0.516, 0.633, 0.636, 0.653, ...   \n",
       "4     {'scores': [0.002, 0.055, 0.064, 0.028, 0.027,...   \n",
       "...                                                 ...   \n",
       "8406  {'scores': [0.002, 0.071, 0.197, 0.187, 0.134,...   \n",
       "8407  {'scores': [0.002, 0.034, 0.028, 0.023, 0.034,...   \n",
       "8408  {'scores': [0.003, 0.363, 0.592, 0.684, 0.671,...   \n",
       "8409  {'scores': [0.002, 0.088, 0.092, 0.107, 0.118,...   \n",
       "8410  {'scores': [0.002, 0.054, 0.041, 0.032, 0.054,...   \n",
       "\n",
       "                                   prediction_coil_fess  \\\n",
       "0     {'scores': [0.99, 0.478, 0.296, 0.229, 0.237, ...   \n",
       "1     {'scores': [0.995, 0.927, 0.919, 0.952, 0.97, ...   \n",
       "2     {'scores': [0.995, 0.915, 0.912, 0.874, 0.805,...   \n",
       "3     {'scores': [0.987, 0.467, 0.348, 0.341, 0.32, ...   \n",
       "4     {'scores': [0.995, 0.933, 0.923, 0.962, 0.935,...   \n",
       "...                                                 ...   \n",
       "8406  {'scores': [0.995, 0.89, 0.684, 0.666, 0.709, ...   \n",
       "8407  {'scores': [0.995, 0.951, 0.914, 0.942, 0.914,...   \n",
       "8408  {'scores': [0.994, 0.609, 0.367, 0.253, 0.249,...   \n",
       "8409  {'scores': [0.994, 0.746, 0.711, 0.694, 0.76, ...   \n",
       "8410  {'scores': [0.995, 0.889, 0.748, 0.622, 0.482,...   \n",
       "\n",
       "                                       score_helix_fess  \n",
       "0     [0.004, 0.068, 0.114, 0.153, 0.145, 0.125, 0.1...  \n",
       "1     [0.003, 0.028, 0.03, 0.011, 0.013, 0.054, 0.12...  \n",
       "2     [0.003, 0.021, 0.02, 0.018, 0.054, 0.082, 0.12...  \n",
       "3     [0.003, 0.018, 0.019, 0.023, 0.027, 0.03, 0.03...  \n",
       "4     [0.003, 0.011, 0.013, 0.01, 0.038, 0.036, 0.03...  \n",
       "...                                                 ...  \n",
       "8406  [0.003, 0.039, 0.119, 0.146, 0.157, 0.236, 0.2...  \n",
       "8407  [0.003, 0.015, 0.058, 0.035, 0.052, 0.05, 0.06...  \n",
       "8408  [0.003, 0.028, 0.041, 0.063, 0.08, 0.126, 0.11...  \n",
       "8409  [0.004, 0.167, 0.198, 0.199, 0.121, 0.046, 0.0...  \n",
       "8410  [0.003, 0.057, 0.21, 0.346, 0.464, 0.49, 0.637...  \n",
       "\n",
       "[8411 rows x 6 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mobidb_lite = mobidb_lite[['acc', 'length', 'prediction_helix_fess', 'prediction_sheet_fess', 'prediction_coil_fess', 'score_helix_fess']].reset_index()\n",
    "mobidb_lite.drop(columns='index', inplace= True)\n",
    "mobidb_lite"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "mobidb_lite['score_helix_fess'] = mobidb_lite.prediction_helix_fess.apply(lambda x: x['scores'])\n",
    "mobidb_lite['score_sheet_fess'] = mobidb_lite.prediction_sheet_fess.apply(lambda x: x['scores'])\n",
    "mobidb_lite['score_coil_fess'] = mobidb_lite.prediction_coil_fess.apply(lambda x: x['scores'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>acc</th>\n",
       "      <th>length</th>\n",
       "      <th>prediction_helix_fess</th>\n",
       "      <th>prediction_sheet_fess</th>\n",
       "      <th>prediction_coil_fess</th>\n",
       "      <th>score_helix_fess</th>\n",
       "      <th>score_sheet_fess</th>\n",
       "      <th>score_coil_fess</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A091DQI4</td>\n",
       "      <td>973</td>\n",
       "      <td>{'scores': [0.004, 0.068, 0.114, 0.153, 0.145,...</td>\n",
       "      <td>{'scores': [0.006, 0.454, 0.59, 0.617, 0.618, ...</td>\n",
       "      <td>{'scores': [0.99, 0.478, 0.296, 0.229, 0.237, ...</td>\n",
       "      <td>[0.004, 0.068, 0.114, 0.153, 0.145, 0.125, 0.1...</td>\n",
       "      <td>[0.006, 0.454, 0.59, 0.617, 0.618, 0.535, 0.37...</td>\n",
       "      <td>[0.99, 0.478, 0.296, 0.229, 0.237, 0.341, 0.51...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0A0A2V081</td>\n",
       "      <td>315</td>\n",
       "      <td>{'scores': [0.003, 0.028, 0.03, 0.011, 0.013, ...</td>\n",
       "      <td>{'scores': [0.002, 0.044, 0.051, 0.037, 0.016,...</td>\n",
       "      <td>{'scores': [0.995, 0.927, 0.919, 0.952, 0.97, ...</td>\n",
       "      <td>[0.003, 0.028, 0.03, 0.011, 0.013, 0.054, 0.12...</td>\n",
       "      <td>[0.002, 0.044, 0.051, 0.037, 0.016, 0.015, 0.0...</td>\n",
       "      <td>[0.995, 0.927, 0.919, 0.952, 0.97, 0.931, 0.85...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0A0D9R447</td>\n",
       "      <td>433</td>\n",
       "      <td>{'scores': [0.003, 0.021, 0.02, 0.018, 0.054, ...</td>\n",
       "      <td>{'scores': [0.002, 0.064, 0.068, 0.107, 0.141,...</td>\n",
       "      <td>{'scores': [0.995, 0.915, 0.912, 0.874, 0.805,...</td>\n",
       "      <td>[0.003, 0.021, 0.02, 0.018, 0.054, 0.082, 0.12...</td>\n",
       "      <td>[0.002, 0.064, 0.068, 0.107, 0.141, 0.213, 0.2...</td>\n",
       "      <td>[0.995, 0.915, 0.912, 0.874, 0.805, 0.706, 0.5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A0A0D9RZF3</td>\n",
       "      <td>974</td>\n",
       "      <td>{'scores': [0.003, 0.018, 0.019, 0.023, 0.027,...</td>\n",
       "      <td>{'scores': [0.01, 0.516, 0.633, 0.636, 0.653, ...</td>\n",
       "      <td>{'scores': [0.987, 0.467, 0.348, 0.341, 0.32, ...</td>\n",
       "      <td>[0.003, 0.018, 0.019, 0.023, 0.027, 0.03, 0.03...</td>\n",
       "      <td>[0.01, 0.516, 0.633, 0.636, 0.653, 0.61, 0.437...</td>\n",
       "      <td>[0.987, 0.467, 0.348, 0.341, 0.32, 0.359, 0.52...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A0A0E0IYS8</td>\n",
       "      <td>513</td>\n",
       "      <td>{'scores': [0.003, 0.011, 0.013, 0.01, 0.038, ...</td>\n",
       "      <td>{'scores': [0.002, 0.055, 0.064, 0.028, 0.027,...</td>\n",
       "      <td>{'scores': [0.995, 0.933, 0.923, 0.962, 0.935,...</td>\n",
       "      <td>[0.003, 0.011, 0.013, 0.01, 0.038, 0.036, 0.03...</td>\n",
       "      <td>[0.002, 0.055, 0.064, 0.028, 0.027, 0.031, 0.0...</td>\n",
       "      <td>[0.995, 0.933, 0.923, 0.962, 0.935, 0.933, 0.9...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8406</th>\n",
       "      <td>D6W6N7</td>\n",
       "      <td>359</td>\n",
       "      <td>{'scores': [0.003, 0.039, 0.119, 0.146, 0.157,...</td>\n",
       "      <td>{'scores': [0.002, 0.071, 0.197, 0.187, 0.134,...</td>\n",
       "      <td>{'scores': [0.995, 0.89, 0.684, 0.666, 0.709, ...</td>\n",
       "      <td>[0.003, 0.039, 0.119, 0.146, 0.157, 0.236, 0.2...</td>\n",
       "      <td>[0.002, 0.071, 0.197, 0.187, 0.134, 0.142, 0.1...</td>\n",
       "      <td>[0.995, 0.89, 0.684, 0.666, 0.709, 0.622, 0.58...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8407</th>\n",
       "      <td>G5EA63</td>\n",
       "      <td>395</td>\n",
       "      <td>{'scores': [0.003, 0.015, 0.058, 0.035, 0.052,...</td>\n",
       "      <td>{'scores': [0.002, 0.034, 0.028, 0.023, 0.034,...</td>\n",
       "      <td>{'scores': [0.995, 0.951, 0.914, 0.942, 0.914,...</td>\n",
       "      <td>[0.003, 0.015, 0.058, 0.035, 0.052, 0.05, 0.06...</td>\n",
       "      <td>[0.002, 0.034, 0.028, 0.023, 0.034, 0.064, 0.0...</td>\n",
       "      <td>[0.995, 0.951, 0.914, 0.942, 0.914, 0.887, 0.8...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8408</th>\n",
       "      <td>H0Z3Y5</td>\n",
       "      <td>1130</td>\n",
       "      <td>{'scores': [0.003, 0.028, 0.041, 0.063, 0.08, ...</td>\n",
       "      <td>{'scores': [0.003, 0.363, 0.592, 0.684, 0.671,...</td>\n",
       "      <td>{'scores': [0.994, 0.609, 0.367, 0.253, 0.249,...</td>\n",
       "      <td>[0.003, 0.028, 0.041, 0.063, 0.08, 0.126, 0.11...</td>\n",
       "      <td>[0.003, 0.363, 0.592, 0.684, 0.671, 0.577, 0.4...</td>\n",
       "      <td>[0.994, 0.609, 0.367, 0.253, 0.249, 0.297, 0.4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8409</th>\n",
       "      <td>P36507</td>\n",
       "      <td>400</td>\n",
       "      <td>{'scores': [0.004, 0.167, 0.198, 0.199, 0.121,...</td>\n",
       "      <td>{'scores': [0.002, 0.088, 0.092, 0.107, 0.118,...</td>\n",
       "      <td>{'scores': [0.994, 0.746, 0.711, 0.694, 0.76, ...</td>\n",
       "      <td>[0.004, 0.167, 0.198, 0.199, 0.121, 0.046, 0.0...</td>\n",
       "      <td>[0.002, 0.088, 0.092, 0.107, 0.118, 0.08, 0.09...</td>\n",
       "      <td>[0.994, 0.746, 0.711, 0.694, 0.76, 0.873, 0.86...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8410</th>\n",
       "      <td>U3JUJ8</td>\n",
       "      <td>1150</td>\n",
       "      <td>{'scores': [0.003, 0.057, 0.21, 0.346, 0.464, ...</td>\n",
       "      <td>{'scores': [0.002, 0.054, 0.041, 0.032, 0.054,...</td>\n",
       "      <td>{'scores': [0.995, 0.889, 0.748, 0.622, 0.482,...</td>\n",
       "      <td>[0.003, 0.057, 0.21, 0.346, 0.464, 0.49, 0.637...</td>\n",
       "      <td>[0.002, 0.054, 0.041, 0.032, 0.054, 0.103, 0.0...</td>\n",
       "      <td>[0.995, 0.889, 0.748, 0.622, 0.482, 0.407, 0.2...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8411 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             acc  length                              prediction_helix_fess  \\\n",
       "0     A0A091DQI4     973  {'scores': [0.004, 0.068, 0.114, 0.153, 0.145,...   \n",
       "1     A0A0A2V081     315  {'scores': [0.003, 0.028, 0.03, 0.011, 0.013, ...   \n",
       "2     A0A0D9R447     433  {'scores': [0.003, 0.021, 0.02, 0.018, 0.054, ...   \n",
       "3     A0A0D9RZF3     974  {'scores': [0.003, 0.018, 0.019, 0.023, 0.027,...   \n",
       "4     A0A0E0IYS8     513  {'scores': [0.003, 0.011, 0.013, 0.01, 0.038, ...   \n",
       "...          ...     ...                                                ...   \n",
       "8406      D6W6N7     359  {'scores': [0.003, 0.039, 0.119, 0.146, 0.157,...   \n",
       "8407      G5EA63     395  {'scores': [0.003, 0.015, 0.058, 0.035, 0.052,...   \n",
       "8408      H0Z3Y5    1130  {'scores': [0.003, 0.028, 0.041, 0.063, 0.08, ...   \n",
       "8409      P36507     400  {'scores': [0.004, 0.167, 0.198, 0.199, 0.121,...   \n",
       "8410      U3JUJ8    1150  {'scores': [0.003, 0.057, 0.21, 0.346, 0.464, ...   \n",
       "\n",
       "                                  prediction_sheet_fess  \\\n",
       "0     {'scores': [0.006, 0.454, 0.59, 0.617, 0.618, ...   \n",
       "1     {'scores': [0.002, 0.044, 0.051, 0.037, 0.016,...   \n",
       "2     {'scores': [0.002, 0.064, 0.068, 0.107, 0.141,...   \n",
       "3     {'scores': [0.01, 0.516, 0.633, 0.636, 0.653, ...   \n",
       "4     {'scores': [0.002, 0.055, 0.064, 0.028, 0.027,...   \n",
       "...                                                 ...   \n",
       "8406  {'scores': [0.002, 0.071, 0.197, 0.187, 0.134,...   \n",
       "8407  {'scores': [0.002, 0.034, 0.028, 0.023, 0.034,...   \n",
       "8408  {'scores': [0.003, 0.363, 0.592, 0.684, 0.671,...   \n",
       "8409  {'scores': [0.002, 0.088, 0.092, 0.107, 0.118,...   \n",
       "8410  {'scores': [0.002, 0.054, 0.041, 0.032, 0.054,...   \n",
       "\n",
       "                                   prediction_coil_fess  \\\n",
       "0     {'scores': [0.99, 0.478, 0.296, 0.229, 0.237, ...   \n",
       "1     {'scores': [0.995, 0.927, 0.919, 0.952, 0.97, ...   \n",
       "2     {'scores': [0.995, 0.915, 0.912, 0.874, 0.805,...   \n",
       "3     {'scores': [0.987, 0.467, 0.348, 0.341, 0.32, ...   \n",
       "4     {'scores': [0.995, 0.933, 0.923, 0.962, 0.935,...   \n",
       "...                                                 ...   \n",
       "8406  {'scores': [0.995, 0.89, 0.684, 0.666, 0.709, ...   \n",
       "8407  {'scores': [0.995, 0.951, 0.914, 0.942, 0.914,...   \n",
       "8408  {'scores': [0.994, 0.609, 0.367, 0.253, 0.249,...   \n",
       "8409  {'scores': [0.994, 0.746, 0.711, 0.694, 0.76, ...   \n",
       "8410  {'scores': [0.995, 0.889, 0.748, 0.622, 0.482,...   \n",
       "\n",
       "                                       score_helix_fess  \\\n",
       "0     [0.004, 0.068, 0.114, 0.153, 0.145, 0.125, 0.1...   \n",
       "1     [0.003, 0.028, 0.03, 0.011, 0.013, 0.054, 0.12...   \n",
       "2     [0.003, 0.021, 0.02, 0.018, 0.054, 0.082, 0.12...   \n",
       "3     [0.003, 0.018, 0.019, 0.023, 0.027, 0.03, 0.03...   \n",
       "4     [0.003, 0.011, 0.013, 0.01, 0.038, 0.036, 0.03...   \n",
       "...                                                 ...   \n",
       "8406  [0.003, 0.039, 0.119, 0.146, 0.157, 0.236, 0.2...   \n",
       "8407  [0.003, 0.015, 0.058, 0.035, 0.052, 0.05, 0.06...   \n",
       "8408  [0.003, 0.028, 0.041, 0.063, 0.08, 0.126, 0.11...   \n",
       "8409  [0.004, 0.167, 0.198, 0.199, 0.121, 0.046, 0.0...   \n",
       "8410  [0.003, 0.057, 0.21, 0.346, 0.464, 0.49, 0.637...   \n",
       "\n",
       "                                       score_sheet_fess  \\\n",
       "0     [0.006, 0.454, 0.59, 0.617, 0.618, 0.535, 0.37...   \n",
       "1     [0.002, 0.044, 0.051, 0.037, 0.016, 0.015, 0.0...   \n",
       "2     [0.002, 0.064, 0.068, 0.107, 0.141, 0.213, 0.2...   \n",
       "3     [0.01, 0.516, 0.633, 0.636, 0.653, 0.61, 0.437...   \n",
       "4     [0.002, 0.055, 0.064, 0.028, 0.027, 0.031, 0.0...   \n",
       "...                                                 ...   \n",
       "8406  [0.002, 0.071, 0.197, 0.187, 0.134, 0.142, 0.1...   \n",
       "8407  [0.002, 0.034, 0.028, 0.023, 0.034, 0.064, 0.0...   \n",
       "8408  [0.003, 0.363, 0.592, 0.684, 0.671, 0.577, 0.4...   \n",
       "8409  [0.002, 0.088, 0.092, 0.107, 0.118, 0.08, 0.09...   \n",
       "8410  [0.002, 0.054, 0.041, 0.032, 0.054, 0.103, 0.0...   \n",
       "\n",
       "                                        score_coil_fess  \n",
       "0     [0.99, 0.478, 0.296, 0.229, 0.237, 0.341, 0.51...  \n",
       "1     [0.995, 0.927, 0.919, 0.952, 0.97, 0.931, 0.85...  \n",
       "2     [0.995, 0.915, 0.912, 0.874, 0.805, 0.706, 0.5...  \n",
       "3     [0.987, 0.467, 0.348, 0.341, 0.32, 0.359, 0.52...  \n",
       "4     [0.995, 0.933, 0.923, 0.962, 0.935, 0.933, 0.9...  \n",
       "...                                                 ...  \n",
       "8406  [0.995, 0.89, 0.684, 0.666, 0.709, 0.622, 0.58...  \n",
       "8407  [0.995, 0.951, 0.914, 0.942, 0.914, 0.887, 0.8...  \n",
       "8408  [0.994, 0.609, 0.367, 0.253, 0.249, 0.297, 0.4...  \n",
       "8409  [0.994, 0.746, 0.711, 0.694, 0.76, 0.873, 0.86...  \n",
       "8410  [0.995, 0.889, 0.748, 0.622, 0.482, 0.407, 0.2...  \n",
       "\n",
       "[8411 rows x 8 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mobidb_lite # ok!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Control - the length of the score lists must be equal to the protein length (one score per position)\n",
    "# (mobidb_lite.score_helix_fess.apply(lambda x: len(x)) == mobidb_lite.length).value_counts() # ok!\n",
    "# (mobidb_lite.score_sheet_fess.apply(lambda x: len(x)) == mobidb_lite.length).value_counts() # ok!\n",
    "# (mobidb_lite.score_coil_fess.apply(lambda x: len(x)) == mobidb_lite.length).value_counts() # Ok!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "mobidb_lite.rename(columns= {'acc': 'uniprot_acc'}, inplace= True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "mobidb_sec_scores = mobidb_lite[['uniprot_acc', 'length', 'score_helix_fess', 'score_sheet_fess', 'score_coil_fess']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mobidb_sec_scores.to_csv('../datasets/mobidb_secondary_scores.tsv', sep= '\\t', index= False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
